## Обнаружение статистически значимых отличий в уровнях экспрессии генов больных раком

Данные для этой задачи взяты из исследования, проведенного в Stanford School of Medicine. В исследовании была предпринята попытка выявить набор  генов, которые позволили  бы более точно диагностировать возникновение рака груди на самых ранних стадиях.

В эксперименте принимали участие 24 человек, у которых не было рака груди (normal), 25 человек, у которых это заболевание было диагностировано на ранней стадии (early neoplasia), и 23 человека с сильно выраженными симптомами (cancer).
Ученые провели секвенирование биологического материала испытуемых, чтобы понять, какие из этих генов наиболее активны в клетках больных людей. В данных представленна именно эта количественная мера активности каждого из 15748 генов у каждого из 72 человек, принимавших участие в эксперименте.

Нужно будет определить те гены, активность которых у людей в разных стадиях заболевания отличается статистически значимо.

In [105]:
import pandas as pd
import numpy as np
import scipy
import statsmodels
import statistics
from statsmodels.stats.weightstats import *
import statsmodels.stats.multitest as smm

In [8]:
def write_answer(file_name, answer):
    with open(file_name, 'w') as fout:
        fout.write(str(answer))

### Часть 1: применение t-критерия Стьюдента

Для начала дважды применим критерий Стьюдента для проверки гипотезы о равенстве средних в двух независимых выборках.

Критерий применим:
* для групп normal (control) и early neoplasia (treatment)
* для групп early neoplasia (control) и cancer (treatment)

Ответом в этой части определим количество статистически значимых отличий, которые будут найдены с помощью t-критерия Стьюдента, то есть число генов, у которых p-value этого теста оказался меньше, чем уровень значимости.

In [9]:
data = pd.read_csv('gene_high_throughput_sequencing.csv', sep=',', header=0)

In [10]:
data.head()

Patient_id Diagnosis  LOC643837  LOC100130417     SAMD11  \
0  STT5425_Breast_001_normal    normal   1.257614      2.408148  13.368622   
1  STT5427_Breast_023_normal    normal   4.567931     16.602734  42.477752   
2  STT5430_Breast_002_normal    normal   2.077597      3.978294  12.863214   
3  STT5439_Breast_003_normal    normal   2.066576      8.520713  14.466035   
4  STT5441_Breast_004_normal    normal   2.613616      3.434965  12.682222   

       NOC2L     KLHL17    PLEKHN1   C1orf170       HES4  ...      CLIC2  \
0   9.494779  20.880435  12.722017   9.494779  54.349694  ...   4.761250   
1  25.562376  23.221137  11.622386  14.330573  72.445474  ...   6.871902   
2  13.728915  14.543176  14.141907   6.232790  57.011005  ...   7.096343   
3   7.823932   8.520713   2.066576  10.870009  53.292034  ...   5.200770   
4  10.543189  26.688686  12.484822   1.364917  67.140393  ...  11.227770   

     RPS4Y1       ZFY      PRKY     USP9Y     DDX3Y       CD24  CYorf15B  \
0  1.257614  1.257614  1.257614  1.257614  1.257614  23.268694  1.257614   
1  1.815112  1.815112  1.815112  1.815112  1.815112  10.427023  1.815112   
2  2.077597  2.077597  2.077597  2.077597  2.077597  22.344226  2.077597   
3  2.066576  2.066576  2.066576  2.066576  2.066576  49.295538  2.066576   
4  1.364917  1.364917  1.364917  1.364917  1.364917  23.627911  1.364917   

      KDM5D    EIF1AY  
0  1.257614  1.257614  
1  1.815112  1.815112  
2  2.077597  2.077597  
3  2.066576  2.066576  
4  1.364917  1.364917  

[5 rows x 15750 columns]

In [7]:
data.shape

(72, 15750)

In [13]:
data['Diagnosis'].value_counts()

early neoplasia    25
normal             24
cancer             23
Name: Diagnosis, dtype: int64

In [14]:
early_neoplasia = data[data['Diagnosis']== 'early neoplasia'].drop(['Patient_id', 'Diagnosis'], axis=1)
normal = data[data['Diagnosis'] == 'normal'].drop(['Patient_id', 'Diagnosis'], axis=1)
cancer = data[data['Diagnosis'] == 'cancer'].drop(['Patient_id', 'Diagnosis'], axis=1)
data_columns = data.columns.drop(['Patient_id', 'Diagnosis'])

In [25]:
print('Shapiro-Wilk normality test, W-statistic:')
for i in data_columns:
    print('\tNormal neoplasia \'' + col_name + '\': %f, p-value: %f' % stats.shapiro(normal[i]))
    print('\tEarly neoplasia \'' + col_name + '\': %f, p-value: %f' % stats.shapiro(early_neoplasia[i]))
    print('\tCancer  \'' + col_name + '\': %f, p-value: %f' % stats.shapiro(cancer[i]))

# Вывод: при уровне значимости 0.05 не все данные распределены нормально

Shapiro-Wilk normality test, W-statistic:
	Normal neoplasia 'EIF1AY': 0.895629, p-value: 0.017408
	Early neoplasia 'EIF1AY': 0.897122, p-value: 0.015890
	Cancer  'EIF1AY': 0.853136, p-value: 0.003103
	Normal neoplasia 'EIF1AY': 0.670455, p-value: 0.000004
	Early neoplasia 'EIF1AY': 0.946906, p-value: 0.213321
	Cancer  'EIF1AY': 0.904603, p-value: 0.031487
	Normal neoplasia 'EIF1AY': 0.724633, p-value: 0.000022
	Early neoplasia 'EIF1AY': 0.930378, p-value: 0.088673
	Cancer  'EIF1AY': 0.843589, p-value: 0.002086
	Normal neoplasia 'EIF1AY': 0.920615, p-value: 0.060266
	Early neoplasia 'EIF1AY': 0.972330, p-value: 0.704477
	Cancer  'EIF1AY': 0.890680, p-value: 0.016324
	Normal neoplasia 'EIF1AY': 0.897267, p-value: 0.018844
	Early neoplasia 'EIF1AY': 0.985678, p-value: 0.970186
	Cancer  'EIF1AY': 0.940256, p-value: 0.181999
	Normal neoplasia 'EIF1AY': 0.929567, p-value: 0.095339
	Early neoplasia 'EIF1AY': 0.970829, p-value: 0.666147
	Cancer  'EIF1AY': 0.881133, p-value: 0.010537
	Normal ne

	Normal neoplasia 'EIF1AY': 0.941563, p-value: 0.176738
	Early neoplasia 'EIF1AY': 0.968691, p-value: 0.612074
	Cancer  'EIF1AY': 0.957226, p-value: 0.409708
	Normal neoplasia 'EIF1AY': 0.861553, p-value: 0.003610
	Early neoplasia 'EIF1AY': 0.953816, p-value: 0.305057
	Cancer  'EIF1AY': 0.972871, p-value: 0.757274
	Normal neoplasia 'EIF1AY': 0.938671, p-value: 0.152356
	Early neoplasia 'EIF1AY': 0.946133, p-value: 0.204831
	Cancer  'EIF1AY': 0.959404, p-value: 0.451434
	Normal neoplasia 'EIF1AY': 0.966563, p-value: 0.583331
	Early neoplasia 'EIF1AY': 0.881874, p-value: 0.007556
	Cancer  'EIF1AY': 0.968708, p-value: 0.658095
	Normal neoplasia 'EIF1AY': 0.967170, p-value: 0.597860
	Early neoplasia 'EIF1AY': 0.986990, p-value: 0.981237
	Cancer  'EIF1AY': 0.950610, p-value: 0.301267
	Normal neoplasia 'EIF1AY': 0.734645, p-value: 0.000030
	Early neoplasia 'EIF1AY': 0.906213, p-value: 0.025112
	Cancer  'EIF1AY': 0.929382, p-value: 0.106055
	Normal neoplasia 'EIF1AY': 0.941859, p-value: 0.179

	Early neoplasia 'EIF1AY': 0.911879, p-value: 0.033576
	Cancer  'EIF1AY': 0.951507, p-value: 0.314377
	Normal neoplasia 'EIF1AY': 0.970431, p-value: 0.677633
	Early neoplasia 'EIF1AY': 0.969945, p-value: 0.643663
	Cancer  'EIF1AY': 0.914342, p-value: 0.050460
	Normal neoplasia 'EIF1AY': 0.912272, p-value: 0.039519
	Early neoplasia 'EIF1AY': 0.845703, p-value: 0.001461
	Cancer  'EIF1AY': 0.777262, p-value: 0.000169
	Normal neoplasia 'EIF1AY': 0.973431, p-value: 0.751514
	Early neoplasia 'EIF1AY': 0.960348, p-value: 0.421440
	Cancer  'EIF1AY': 0.953975, p-value: 0.352988
	Normal neoplasia 'EIF1AY': 0.893576, p-value: 0.015770
	Early neoplasia 'EIF1AY': 0.934380, p-value: 0.109733
	Cancer  'EIF1AY': 0.812041, p-value: 0.000601
	Normal neoplasia 'EIF1AY': 0.951467, p-value: 0.291421
	Early neoplasia 'EIF1AY': 0.894300, p-value: 0.013815
	Cancer  'EIF1AY': 0.883923, p-value: 0.011962
	Normal neoplasia 'EIF1AY': 0.895553, p-value: 0.017345
	Early neoplasia 'EIF1AY': 0.757219, p-value: 0.0000

	Normal neoplasia 'EIF1AY': 0.958145, p-value: 0.402301
	Early neoplasia 'EIF1AY': 0.989748, p-value: 0.994871
	Cancer  'EIF1AY': 0.967687, p-value: 0.633815
	Normal neoplasia 'EIF1AY': 0.988208, p-value: 0.990242
	Early neoplasia 'EIF1AY': 0.968323, p-value: 0.602886
	Cancer  'EIF1AY': 0.935897, p-value: 0.146642
	Normal neoplasia 'EIF1AY': 0.979025, p-value: 0.877231
	Early neoplasia 'EIF1AY': 0.977475, p-value: 0.830943
	Cancer  'EIF1AY': 0.967290, p-value: 0.624452
	Normal neoplasia 'EIF1AY': 0.902124, p-value: 0.023881
	Early neoplasia 'EIF1AY': 0.949714, p-value: 0.247027
	Cancer  'EIF1AY': 0.964326, p-value: 0.555991
	Normal neoplasia 'EIF1AY': 0.949621, p-value: 0.265877
	Early neoplasia 'EIF1AY': 0.945389, p-value: 0.196956
	Cancer  'EIF1AY': 0.948873, p-value: 0.277244
	Normal neoplasia 'EIF1AY': 0.958965, p-value: 0.417978
	Early neoplasia 'EIF1AY': 0.970347, p-value: 0.653872
	Cancer  'EIF1AY': 0.963743, p-value: 0.542926
	Normal neoplasia 'EIF1AY': 0.944252, p-value: 0.202

	Cancer  'EIF1AY': 0.871692, p-value: 0.006907
	Normal neoplasia 'EIF1AY': 0.934840, p-value: 0.125089
	Early neoplasia 'EIF1AY': 0.956575, p-value: 0.350495
	Cancer  'EIF1AY': 0.947764, p-value: 0.262816
	Normal neoplasia 'EIF1AY': 0.974232, p-value: 0.770864
	Early neoplasia 'EIF1AY': 0.915019, p-value: 0.039504
	Cancer  'EIF1AY': 0.774379, p-value: 0.000153
	Normal neoplasia 'EIF1AY': 0.959329, p-value: 0.425097
	Early neoplasia 'EIF1AY': 0.925253, p-value: 0.067566
	Cancer  'EIF1AY': 0.927573, p-value: 0.096934
	Normal neoplasia 'EIF1AY': 0.947466, p-value: 0.238614
	Early neoplasia 'EIF1AY': 0.943774, p-value: 0.180851
	Cancer  'EIF1AY': 0.942148, p-value: 0.199804
	Normal neoplasia 'EIF1AY': 0.883807, p-value: 0.009922
	Early neoplasia 'EIF1AY': 0.911264, p-value: 0.032529
	Cancer  'EIF1AY': 0.973031, p-value: 0.761038
	Normal neoplasia 'EIF1AY': 0.964393, p-value: 0.532765
	Early neoplasia 'EIF1AY': 0.887734, p-value: 0.010018
	Cancer  'EIF1AY': 0.922932, p-value: 0.077004
	Norm

	Early neoplasia 'EIF1AY': 0.732290, p-value: 0.000020
	Cancer  'EIF1AY': 0.926790, p-value: 0.093237
	Normal neoplasia 'EIF1AY': 0.939972, p-value: 0.162889
	Early neoplasia 'EIF1AY': 0.981301, p-value: 0.909619
	Cancer  'EIF1AY': 0.970944, p-value: 0.711589
	Normal neoplasia 'EIF1AY': 0.833389, p-value: 0.001092
	Early neoplasia 'EIF1AY': 0.903322, p-value: 0.021686
	Cancer  'EIF1AY': 0.909816, p-value: 0.040484
	Normal neoplasia 'EIF1AY': 0.957778, p-value: 0.395432
	Early neoplasia 'EIF1AY': 0.938024, p-value: 0.133249
	Cancer  'EIF1AY': 0.866957, p-value: 0.005610
	Normal neoplasia 'EIF1AY': 0.952472, p-value: 0.306236
	Early neoplasia 'EIF1AY': 0.914283, p-value: 0.038022
	Cancer  'EIF1AY': 0.975327, p-value: 0.813407
	Normal neoplasia 'EIF1AY': 0.918410, p-value: 0.053875
	Early neoplasia 'EIF1AY': 0.985465, p-value: 0.968084
	Cancer  'EIF1AY': 0.964020, p-value: 0.549110
	Normal neoplasia 'EIF1AY': 0.919277, p-value: 0.056301
	Early neoplasia 'EIF1AY': 0.956184, p-value: 0.3437

	Cancer  'EIF1AY': 0.650225, p-value: 0.000003
	Normal neoplasia 'EIF1AY': 0.928372, p-value: 0.089653
	Early neoplasia 'EIF1AY': 0.958106, p-value: 0.378035
	Cancer  'EIF1AY': 0.966342, p-value: 0.602222
	Normal neoplasia 'EIF1AY': 0.959405, p-value: 0.426577
	Early neoplasia 'EIF1AY': 0.963154, p-value: 0.480805
	Cancer  'EIF1AY': 0.973740, p-value: 0.777509
	Normal neoplasia 'EIF1AY': 0.982170, p-value: 0.932218
	Early neoplasia 'EIF1AY': 0.951540, p-value: 0.271507
	Cancer  'EIF1AY': 0.987560, p-value: 0.988786
	Normal neoplasia 'EIF1AY': 0.979683, p-value: 0.889948
	Early neoplasia 'EIF1AY': 0.966781, p-value: 0.565012
	Cancer  'EIF1AY': 0.961555, p-value: 0.495472
	Normal neoplasia 'EIF1AY': 0.944677, p-value: 0.207181
	Early neoplasia 'EIF1AY': 0.979968, p-value: 0.884476
	Cancer  'EIF1AY': 0.599614, p-value: 0.000001
	Normal neoplasia 'EIF1AY': 0.957943, p-value: 0.398500
	Early neoplasia 'EIF1AY': 0.961803, p-value: 0.451534
	Cancer  'EIF1AY': 0.947728, p-value: 0.262358
	Norm

	Normal neoplasia 'EIF1AY': 0.938174, p-value: 0.148514
	Early neoplasia 'EIF1AY': 0.909165, p-value: 0.029201
	Cancer  'EIF1AY': 0.940706, p-value: 0.186096
	Normal neoplasia 'EIF1AY': 0.831892, p-value: 0.001027
	Early neoplasia 'EIF1AY': 0.974987, p-value: 0.771463
	Cancer  'EIF1AY': 0.900761, p-value: 0.026210
	Normal neoplasia 'EIF1AY': 0.911827, p-value: 0.038647
	Early neoplasia 'EIF1AY': 0.783856, p-value: 0.000124
	Cancer  'EIF1AY': 0.911971, p-value: 0.044950
	Normal neoplasia 'EIF1AY': 0.680683, p-value: 0.000006
	Early neoplasia 'EIF1AY': 0.942360, p-value: 0.167803
	Cancer  'EIF1AY': 0.908151, p-value: 0.037350
	Normal neoplasia 'EIF1AY': 0.940673, p-value: 0.168852
	Early neoplasia 'EIF1AY': 0.929206, p-value: 0.083318
	Cancer  'EIF1AY': 0.950749, p-value: 0.303265
	Normal neoplasia 'EIF1AY': 0.971286, p-value: 0.698800
	Early neoplasia 'EIF1AY': 0.945348, p-value: 0.196538
	Cancer  'EIF1AY': 0.951964, p-value: 0.321243
	Normal neoplasia 'EIF1AY': 0.959459, p-value: 0.427

	Early neoplasia 'EIF1AY': 0.960405, p-value: 0.422601
	Cancer  'EIF1AY': 0.769009, p-value: 0.000127
	Normal neoplasia 'EIF1AY': 0.976804, p-value: 0.830484
	Early neoplasia 'EIF1AY': 0.929780, p-value: 0.085900
	Cancer  'EIF1AY': 0.945918, p-value: 0.240317
	Normal neoplasia 'EIF1AY': 0.912988, p-value: 0.040965
	Early neoplasia 'EIF1AY': 0.943455, p-value: 0.177824
	Cancer  'EIF1AY': 0.920196, p-value: 0.067267
	Normal neoplasia 'EIF1AY': 0.859783, p-value: 0.003339
	Early neoplasia 'EIF1AY': 0.717629, p-value: 0.000013
	Cancer  'EIF1AY': 0.732320, p-value: 0.000038
	Normal neoplasia 'EIF1AY': 0.923550, p-value: 0.070009
	Early neoplasia 'EIF1AY': 0.658255, p-value: 0.000002
	Cancer  'EIF1AY': 0.851500, p-value: 0.002897
	Normal neoplasia 'EIF1AY': 0.870850, p-value: 0.005467
	Early neoplasia 'EIF1AY': 0.843055, p-value: 0.001304
	Cancer  'EIF1AY': 0.851933, p-value: 0.002950
	Normal neoplasia 'EIF1AY': 0.837800, p-value: 0.001309
	Early neoplasia 'EIF1AY': 0.804063, p-value: 0.0002

	Normal neoplasia 'EIF1AY': 0.801809, p-value: 0.000315
	Early neoplasia 'EIF1AY': 0.965799, p-value: 0.541473
	Cancer  'EIF1AY': 0.932646, p-value: 0.124749
	Normal neoplasia 'EIF1AY': 0.941653, p-value: 0.177551
	Early neoplasia 'EIF1AY': 0.793996, p-value: 0.000181
	Cancer  'EIF1AY': 0.938868, p-value: 0.169926
	Normal neoplasia 'EIF1AY': 0.892266, p-value: 0.014809
	Early neoplasia 'EIF1AY': 0.972771, p-value: 0.715705
	Cancer  'EIF1AY': 0.970033, p-value: 0.689806
	Normal neoplasia 'EIF1AY': 0.959249, p-value: 0.423519
	Early neoplasia 'EIF1AY': 0.985749, p-value: 0.970868
	Cancer  'EIF1AY': 0.973923, p-value: 0.781707
	Normal neoplasia 'EIF1AY': 0.977090, p-value: 0.836791
	Early neoplasia 'EIF1AY': 0.952030, p-value: 0.278434
	Cancer  'EIF1AY': 0.962009, p-value: 0.505118
	Normal neoplasia 'EIF1AY': 0.950533, p-value: 0.278229
	Early neoplasia 'EIF1AY': 0.759315, p-value: 0.000051
	Cancer  'EIF1AY': 0.940319, p-value: 0.182566
	Normal neoplasia 'EIF1AY': 0.939098, p-value: 0.155

	Cancer  'EIF1AY': 0.930173, p-value: 0.110312
	Normal neoplasia 'EIF1AY': 0.963171, p-value: 0.505360
	Early neoplasia 'EIF1AY': 0.987921, p-value: 0.987182
	Cancer  'EIF1AY': 0.784134, p-value: 0.000216
	Normal neoplasia 'EIF1AY': 0.937512, p-value: 0.143539
	Early neoplasia 'EIF1AY': 0.936527, p-value: 0.123038
	Cancer  'EIF1AY': 0.949254, p-value: 0.282361
	Normal neoplasia 'EIF1AY': 0.941322, p-value: 0.174568
	Early neoplasia 'EIF1AY': 0.769882, p-value: 0.000074
	Cancer  'EIF1AY': 0.914233, p-value: 0.050193
	Normal neoplasia 'EIF1AY': 0.967794, p-value: 0.612905
	Early neoplasia 'EIF1AY': 0.938041, p-value: 0.133368
	Cancer  'EIF1AY': 0.698473, p-value: 0.000013
	Normal neoplasia 'EIF1AY': 0.955931, p-value: 0.362234
	Early neoplasia 'EIF1AY': 0.943886, p-value: 0.181933
	Cancer  'EIF1AY': 0.971142, p-value: 0.716313
	Normal neoplasia 'EIF1AY': 0.961371, p-value: 0.466593
	Early neoplasia 'EIF1AY': 0.959705, p-value: 0.408636
	Cancer  'EIF1AY': 0.920534, p-value: 0.068398
	Norm

	Cancer  'EIF1AY': 0.951029, p-value: 0.307328
	Normal neoplasia 'EIF1AY': 0.963253, p-value: 0.507171
	Early neoplasia 'EIF1AY': 0.957227, p-value: 0.362016
	Cancer  'EIF1AY': 0.958931, p-value: 0.442118
	Normal neoplasia 'EIF1AY': 0.929382, p-value: 0.094436
	Early neoplasia 'EIF1AY': 0.973443, p-value: 0.732785
	Cancer  'EIF1AY': 0.978932, p-value: 0.887196
	Normal neoplasia 'EIF1AY': 0.909475, p-value: 0.034358
	Early neoplasia 'EIF1AY': 0.886829, p-value: 0.009588
	Cancer  'EIF1AY': 0.834575, p-value: 0.001447
	Normal neoplasia 'EIF1AY': 0.946423, p-value: 0.226376
	Early neoplasia 'EIF1AY': 0.967417, p-value: 0.580512
	Cancer  'EIF1AY': 0.958057, p-value: 0.425277
	Normal neoplasia 'EIF1AY': 0.875317, p-value: 0.006698
	Early neoplasia 'EIF1AY': 0.905550, p-value: 0.024279
	Cancer  'EIF1AY': 0.929977, p-value: 0.109242
	Normal neoplasia 'EIF1AY': 0.936311, p-value: 0.134937
	Early neoplasia 'EIF1AY': 0.936834, p-value: 0.125066
	Cancer  'EIF1AY': 0.849077, p-value: 0.002618
	Norm

	Normal neoplasia 'EIF1AY': 0.939914, p-value: 0.162402
	Early neoplasia 'EIF1AY': 0.915061, p-value: 0.039589
	Cancer  'EIF1AY': 0.715473, p-value: 0.000022
	Normal neoplasia 'EIF1AY': 0.883389, p-value: 0.009730
	Early neoplasia 'EIF1AY': 0.926934, p-value: 0.073855
	Cancer  'EIF1AY': 0.839201, p-value: 0.001744
	Normal neoplasia 'EIF1AY': 0.955725, p-value: 0.358676
	Early neoplasia 'EIF1AY': 0.929967, p-value: 0.086756
	Cancer  'EIF1AY': 0.876703, p-value: 0.008632
	Normal neoplasia 'EIF1AY': 0.965178, p-value: 0.550809
	Early neoplasia 'EIF1AY': 0.945194, p-value: 0.194939
	Cancer  'EIF1AY': 0.986342, p-value: 0.981827
	Normal neoplasia 'EIF1AY': 0.930365, p-value: 0.099337
	Early neoplasia 'EIF1AY': 0.948035, p-value: 0.226322
	Cancer  'EIF1AY': 0.963445, p-value: 0.536332
	Normal neoplasia 'EIF1AY': 0.934230, p-value: 0.121216
	Early neoplasia 'EIF1AY': 0.694412, p-value: 0.000006
	Cancer  'EIF1AY': 0.865049, p-value: 0.005163
	Normal neoplasia 'EIF1AY': 0.960356, p-value: 0.445

	Cancer  'EIF1AY': 0.955798, p-value: 0.383964
	Normal neoplasia 'EIF1AY': 0.753247, p-value: 0.000056
	Early neoplasia 'EIF1AY': 0.795668, p-value: 0.000193
	Cancer  'EIF1AY': 0.941503, p-value: 0.193551
	Normal neoplasia 'EIF1AY': 0.862364, p-value: 0.003741
	Early neoplasia 'EIF1AY': 0.893300, p-value: 0.013151
	Cancer  'EIF1AY': 0.842788, p-value: 0.002019
	Normal neoplasia 'EIF1AY': 0.965602, p-value: 0.560679
	Early neoplasia 'EIF1AY': 0.966554, p-value: 0.559525
	Cancer  'EIF1AY': 0.886744, p-value: 0.013611
	Normal neoplasia 'EIF1AY': 0.977125, p-value: 0.837558
	Early neoplasia 'EIF1AY': 0.955861, p-value: 0.338223
	Cancer  'EIF1AY': 0.943769, p-value: 0.216363
	Normal neoplasia 'EIF1AY': 0.963246, p-value: 0.507029
	Early neoplasia 'EIF1AY': 0.953236, p-value: 0.296176
	Cancer  'EIF1AY': 0.848872, p-value: 0.002596
	Normal neoplasia 'EIF1AY': 0.949179, p-value: 0.260059
	Early neoplasia 'EIF1AY': 0.955382, p-value: 0.330179
	Cancer  'EIF1AY': 0.962873, p-value: 0.523772
	Norm

	Early neoplasia 'EIF1AY': 0.978742, p-value: 0.859085
	Cancer  'EIF1AY': 0.971360, p-value: 0.721514
	Normal neoplasia 'EIF1AY': 0.960416, p-value: 0.446841
	Early neoplasia 'EIF1AY': 0.982343, p-value: 0.927246
	Cancer  'EIF1AY': 0.945799, p-value: 0.238931
	Normal neoplasia 'EIF1AY': 0.888824, p-value: 0.012569
	Early neoplasia 'EIF1AY': 0.883522, p-value: 0.008175
	Cancer  'EIF1AY': 0.947917, p-value: 0.264774
	Normal neoplasia 'EIF1AY': 0.959998, p-value: 0.438375
	Early neoplasia 'EIF1AY': 0.969956, p-value: 0.643966
	Cancer  'EIF1AY': 0.952673, p-value: 0.332152
	Normal neoplasia 'EIF1AY': 0.874452, p-value: 0.006438
	Early neoplasia 'EIF1AY': 0.928937, p-value: 0.082136
	Cancer  'EIF1AY': 0.914394, p-value: 0.050590
	Normal neoplasia 'EIF1AY': 0.933407, p-value: 0.116184
	Early neoplasia 'EIF1AY': 0.945401, p-value: 0.197079
	Cancer  'EIF1AY': 0.980702, p-value: 0.917757
	Normal neoplasia 'EIF1AY': 0.765261, p-value: 0.000084
	Early neoplasia 'EIF1AY': 0.890646, p-value: 0.0115

	Early neoplasia 'EIF1AY': 0.941416, p-value: 0.159606
	Cancer  'EIF1AY': 0.970776, p-value: 0.707576
	Normal neoplasia 'EIF1AY': 0.902206, p-value: 0.023977
	Early neoplasia 'EIF1AY': 0.976419, p-value: 0.806269
	Cancer  'EIF1AY': 0.979018, p-value: 0.888779
	Normal neoplasia 'EIF1AY': 0.973897, p-value: 0.762799
	Early neoplasia 'EIF1AY': 0.853938, p-value: 0.002094
	Cancer  'EIF1AY': 0.941756, p-value: 0.195983
	Normal neoplasia 'EIF1AY': 0.942109, p-value: 0.181739
	Early neoplasia 'EIF1AY': 0.913171, p-value: 0.035894
	Cancer  'EIF1AY': 0.918487, p-value: 0.061837
	Normal neoplasia 'EIF1AY': 0.950474, p-value: 0.277417
	Early neoplasia 'EIF1AY': 0.972673, p-value: 0.713217
	Cancer  'EIF1AY': 0.875412, p-value: 0.008148
	Normal neoplasia 'EIF1AY': 0.980938, p-value: 0.912500
	Early neoplasia 'EIF1AY': 0.963777, p-value: 0.494702
	Cancer  'EIF1AY': 0.979061, p-value: 0.889559
	Normal neoplasia 'EIF1AY': 0.963542, p-value: 0.513595
	Early neoplasia 'EIF1AY': 0.984277, p-value: 0.9547

	Cancer  'EIF1AY': 0.794504, p-value: 0.000313
	Normal neoplasia 'EIF1AY': 0.952520, p-value: 0.306949
	Early neoplasia 'EIF1AY': 0.957929, p-value: 0.374763
	Cancer  'EIF1AY': 0.937907, p-value: 0.162022
	Normal neoplasia 'EIF1AY': 0.928545, p-value: 0.090457
	Early neoplasia 'EIF1AY': 0.947909, p-value: 0.224842
	Cancer  'EIF1AY': 0.893754, p-value: 0.018836
	Normal neoplasia 'EIF1AY': 0.964981, p-value: 0.546245
	Early neoplasia 'EIF1AY': 0.972100, p-value: 0.698619
	Cancer  'EIF1AY': 0.844114, p-value: 0.002132
	Normal neoplasia 'EIF1AY': 0.818662, p-value: 0.000604
	Early neoplasia 'EIF1AY': 0.833785, p-value: 0.000880
	Cancer  'EIF1AY': 0.893648, p-value: 0.018743
	Normal neoplasia 'EIF1AY': 0.963435, p-value: 0.511218
	Early neoplasia 'EIF1AY': 0.946788, p-value: 0.212002
	Cancer  'EIF1AY': 0.895321, p-value: 0.020271
	Normal neoplasia 'EIF1AY': 0.919881, p-value: 0.058057
	Early neoplasia 'EIF1AY': 0.921601, p-value: 0.055727
	Cancer  'EIF1AY': 0.851636, p-value: 0.002913
	Norm

	Early neoplasia 'EIF1AY': 0.947369, p-value: 0.218569
	Cancer  'EIF1AY': 0.962281, p-value: 0.510936
	Normal neoplasia 'EIF1AY': 0.977514, p-value: 0.846009
	Early neoplasia 'EIF1AY': 0.972426, p-value: 0.706929
	Cancer  'EIF1AY': 0.970432, p-value: 0.699337
	Normal neoplasia 'EIF1AY': 0.978868, p-value: 0.874113
	Early neoplasia 'EIF1AY': 0.959047, p-value: 0.395813
	Cancer  'EIF1AY': 0.967401, p-value: 0.627071
	Normal neoplasia 'EIF1AY': 0.902840, p-value: 0.024735
	Early neoplasia 'EIF1AY': 0.955548, p-value: 0.332958
	Cancer  'EIF1AY': 0.948578, p-value: 0.273343
	Normal neoplasia 'EIF1AY': 0.977564, p-value: 0.847077
	Early neoplasia 'EIF1AY': 0.968736, p-value: 0.613208
	Cancer  'EIF1AY': 0.943977, p-value: 0.218581
	Normal neoplasia 'EIF1AY': 0.954032, p-value: 0.330508
	Early neoplasia 'EIF1AY': 0.910998, p-value: 0.032086
	Cancer  'EIF1AY': 0.870140, p-value: 0.006450
	Normal neoplasia 'EIF1AY': 0.945999, p-value: 0.221569
	Early neoplasia 'EIF1AY': 0.892653, p-value: 0.0127

	Normal neoplasia 'EIF1AY': 0.891924, p-value: 0.014569
	Early neoplasia 'EIF1AY': 0.966666, p-value: 0.562230
	Cancer  'EIF1AY': 0.913595, p-value: 0.048652
	Normal neoplasia 'EIF1AY': 0.590626, p-value: 0.000000
	Early neoplasia 'EIF1AY': 0.858475, p-value: 0.002562
	Cancer  'EIF1AY': 0.803259, p-value: 0.000432
	Normal neoplasia 'EIF1AY': 0.931486, p-value: 0.105237
	Early neoplasia 'EIF1AY': 0.983479, p-value: 0.944260
	Cancer  'EIF1AY': 0.940641, p-value: 0.185495
	Normal neoplasia 'EIF1AY': 0.958613, p-value: 0.411197
	Early neoplasia 'EIF1AY': 0.943247, p-value: 0.175877
	Cancer  'EIF1AY': 0.970909, p-value: 0.710750
	Normal neoplasia 'EIF1AY': 0.836781, p-value: 0.001255
	Early neoplasia 'EIF1AY': 0.971177, p-value: 0.675048
	Cancer  'EIF1AY': 0.975896, p-value: 0.825867
	Normal neoplasia 'EIF1AY': 0.936853, p-value: 0.138753
	Early neoplasia 'EIF1AY': 0.940366, p-value: 0.150941
	Cancer  'EIF1AY': 0.967235, p-value: 0.623151
	Normal neoplasia 'EIF1AY': 0.680827, p-value: 0.000

	Cancer  'EIF1AY': 0.895550, p-value: 0.020490
	Normal neoplasia 'EIF1AY': 0.886419, p-value: 0.011218
	Early neoplasia 'EIF1AY': 0.961884, p-value: 0.453260
	Cancer  'EIF1AY': 0.782213, p-value: 0.000202
	Normal neoplasia 'EIF1AY': 0.957699, p-value: 0.393959
	Early neoplasia 'EIF1AY': 0.925870, p-value: 0.069809
	Cancer  'EIF1AY': 0.912895, p-value: 0.047018
	Normal neoplasia 'EIF1AY': 0.956211, p-value: 0.367124
	Early neoplasia 'EIF1AY': 0.954534, p-value: 0.316365
	Cancer  'EIF1AY': 0.924958, p-value: 0.085135
	Normal neoplasia 'EIF1AY': 0.983075, p-value: 0.945056
	Early neoplasia 'EIF1AY': 0.979839, p-value: 0.881905
	Cancer  'EIF1AY': 0.845286, p-value: 0.002237
	Normal neoplasia 'EIF1AY': 0.953335, p-value: 0.319457
	Early neoplasia 'EIF1AY': 0.954304, p-value: 0.312711
	Cancer  'EIF1AY': 0.857184, p-value: 0.003682
	Normal neoplasia 'EIF1AY': 0.925087, p-value: 0.075740
	Early neoplasia 'EIF1AY': 0.955844, p-value: 0.337924
	Cancer  'EIF1AY': 0.889017, p-value: 0.015113
	Norm

	Normal neoplasia 'EIF1AY': 0.962873, p-value: 0.498811
	Early neoplasia 'EIF1AY': 0.674599, p-value: 0.000003
	Cancer  'EIF1AY': 0.932050, p-value: 0.121105
	Normal neoplasia 'EIF1AY': 0.900060, p-value: 0.021586
	Early neoplasia 'EIF1AY': 0.843006, p-value: 0.001301
	Cancer  'EIF1AY': 0.921453, p-value: 0.071573
	Normal neoplasia 'EIF1AY': 0.941683, p-value: 0.177825
	Early neoplasia 'EIF1AY': 0.966638, p-value: 0.561545
	Cancer  'EIF1AY': 0.966021, p-value: 0.594737
	Normal neoplasia 'EIF1AY': 0.967303, p-value: 0.601064
	Early neoplasia 'EIF1AY': 0.942728, p-value: 0.171104
	Cancer  'EIF1AY': 0.985173, p-value: 0.972954
	Normal neoplasia 'EIF1AY': 0.968629, p-value: 0.633252
	Early neoplasia 'EIF1AY': 0.978524, p-value: 0.854377
	Cancer  'EIF1AY': 0.976139, p-value: 0.831128
	Normal neoplasia 'EIF1AY': 0.959784, p-value: 0.434098
	Early neoplasia 'EIF1AY': 0.977568, p-value: 0.833048
	Cancer  'EIF1AY': 0.959631, p-value: 0.455949
	Normal neoplasia 'EIF1AY': 0.702111, p-value: 0.000

	Early neoplasia 'EIF1AY': 0.968736, p-value: 0.613198
	Cancer  'EIF1AY': 0.964548, p-value: 0.560982
	Normal neoplasia 'EIF1AY': 0.971161, p-value: 0.695715
	Early neoplasia 'EIF1AY': 0.946766, p-value: 0.211756
	Cancer  'EIF1AY': 0.875173, p-value: 0.008061
	Normal neoplasia 'EIF1AY': 0.932875, p-value: 0.113045
	Early neoplasia 'EIF1AY': 0.968264, p-value: 0.601440
	Cancer  'EIF1AY': 0.942566, p-value: 0.203957
	Normal neoplasia 'EIF1AY': 0.974816, p-value: 0.784779
	Early neoplasia 'EIF1AY': 0.964884, p-value: 0.519999
	Cancer  'EIF1AY': 0.977839, p-value: 0.866260
	Normal neoplasia 'EIF1AY': 0.938868, p-value: 0.153912
	Early neoplasia 'EIF1AY': 0.918622, p-value: 0.047665
	Cancer  'EIF1AY': 0.976810, p-value: 0.845331
	Normal neoplasia 'EIF1AY': 0.973312, p-value: 0.748614
	Early neoplasia 'EIF1AY': 0.968883, p-value: 0.616892
	Cancer  'EIF1AY': 0.958918, p-value: 0.441867
	Normal neoplasia 'EIF1AY': 0.961336, p-value: 0.465853
	Early neoplasia 'EIF1AY': 0.964147, p-value: 0.5030

	Early neoplasia 'EIF1AY': 0.965448, p-value: 0.533169
	Cancer  'EIF1AY': 0.985368, p-value: 0.974588
	Normal neoplasia 'EIF1AY': 0.975980, p-value: 0.811903
	Early neoplasia 'EIF1AY': 0.950447, p-value: 0.256606
	Cancer  'EIF1AY': 0.963704, p-value: 0.542076
	Normal neoplasia 'EIF1AY': 0.961111, p-value: 0.461148
	Early neoplasia 'EIF1AY': 0.903646, p-value: 0.022045
	Cancer  'EIF1AY': 0.986583, p-value: 0.983386
	Normal neoplasia 'EIF1AY': 0.987769, p-value: 0.988123
	Early neoplasia 'EIF1AY': 0.965842, p-value: 0.542491
	Cancer  'EIF1AY': 0.953185, p-value: 0.340217
	Normal neoplasia 'EIF1AY': 0.978177, p-value: 0.860034
	Early neoplasia 'EIF1AY': 0.933953, p-value: 0.107263
	Cancer  'EIF1AY': 0.981674, p-value: 0.932582
	Normal neoplasia 'EIF1AY': 0.935899, p-value: 0.132102
	Early neoplasia 'EIF1AY': 0.977701, p-value: 0.836090
	Cancer  'EIF1AY': 0.923026, p-value: 0.077361
	Normal neoplasia 'EIF1AY': 0.896723, p-value: 0.018354
	Early neoplasia 'EIF1AY': 0.750167, p-value: 0.0000

	Cancer  'EIF1AY': 0.946348, p-value: 0.245398
	Normal neoplasia 'EIF1AY': 0.981375, p-value: 0.919783
	Early neoplasia 'EIF1AY': 0.939790, p-value: 0.146388
	Cancer  'EIF1AY': 0.974375, p-value: 0.792055
	Normal neoplasia 'EIF1AY': 0.980512, p-value: 0.905107
	Early neoplasia 'EIF1AY': 0.941337, p-value: 0.158934
	Cancer  'EIF1AY': 0.960485, p-value: 0.473220
	Normal neoplasia 'EIF1AY': 0.966140, p-value: 0.573322
	Early neoplasia 'EIF1AY': 0.927676, p-value: 0.076819
	Cancer  'EIF1AY': 0.914874, p-value: 0.051791
	Normal neoplasia 'EIF1AY': 0.966063, p-value: 0.571487
	Early neoplasia 'EIF1AY': 0.934957, p-value: 0.113161
	Cancer  'EIF1AY': 0.968890, p-value: 0.662452
	Normal neoplasia 'EIF1AY': 0.960437, p-value: 0.447266
	Early neoplasia 'EIF1AY': 0.950492, p-value: 0.257209
	Cancer  'EIF1AY': 0.921568, p-value: 0.071981
	Normal neoplasia 'EIF1AY': 0.850499, p-value: 0.002233
	Early neoplasia 'EIF1AY': 0.954793, p-value: 0.320528
	Cancer  'EIF1AY': 0.918499, p-value: 0.061871
	Norm

	Normal neoplasia 'EIF1AY': 0.922069, p-value: 0.064906
	Early neoplasia 'EIF1AY': 0.920359, p-value: 0.052208
	Cancer  'EIF1AY': 0.979660, p-value: 0.900304
	Normal neoplasia 'EIF1AY': 0.977704, p-value: 0.850067
	Early neoplasia 'EIF1AY': 0.952513, p-value: 0.285422
	Cancer  'EIF1AY': 0.947456, p-value: 0.258938
	Normal neoplasia 'EIF1AY': 0.959936, p-value: 0.437141
	Early neoplasia 'EIF1AY': 0.922832, p-value: 0.059459
	Cancer  'EIF1AY': 0.969890, p-value: 0.686371
	Normal neoplasia 'EIF1AY': 0.958799, p-value: 0.414780
	Early neoplasia 'EIF1AY': 0.957170, p-value: 0.360998
	Cancer  'EIF1AY': 0.976712, p-value: 0.843276
	Normal neoplasia 'EIF1AY': 0.968229, p-value: 0.623486
	Early neoplasia 'EIF1AY': 0.946234, p-value: 0.205918
	Cancer  'EIF1AY': 0.962925, p-value: 0.524906
	Normal neoplasia 'EIF1AY': 0.916729, p-value: 0.049474
	Early neoplasia 'EIF1AY': 0.963042, p-value: 0.478317
	Cancer  'EIF1AY': 0.925411, p-value: 0.087068
	Normal neoplasia 'EIF1AY': 0.887059, p-value: 0.011

	Cancer  'EIF1AY': 0.981351, p-value: 0.927825
	Normal neoplasia 'EIF1AY': 0.964077, p-value: 0.525607
	Early neoplasia 'EIF1AY': 0.969186, p-value: 0.624502
	Cancer  'EIF1AY': 0.977554, p-value: 0.860572
	Normal neoplasia 'EIF1AY': 0.954709, p-value: 0.341542
	Early neoplasia 'EIF1AY': 0.985026, p-value: 0.963475
	Cancer  'EIF1AY': 0.983755, p-value: 0.959100
	Normal neoplasia 'EIF1AY': 0.746346, p-value: 0.000044
	Early neoplasia 'EIF1AY': 0.925338, p-value: 0.067870
	Cancer  'EIF1AY': 0.921908, p-value: 0.073202
	Normal neoplasia 'EIF1AY': 0.974904, p-value: 0.786849
	Early neoplasia 'EIF1AY': 0.957301, p-value: 0.363349
	Cancer  'EIF1AY': 0.953767, p-value: 0.349596
	Normal neoplasia 'EIF1AY': 0.802709, p-value: 0.000326
	Early neoplasia 'EIF1AY': 0.893492, p-value: 0.013275
	Cancer  'EIF1AY': 0.942110, p-value: 0.199433
	Normal neoplasia 'EIF1AY': 0.918580, p-value: 0.054342
	Early neoplasia 'EIF1AY': 0.958124, p-value: 0.378383
	Cancer  'EIF1AY': 0.941280, p-value: 0.191436
	Norm

	Normal neoplasia 'EIF1AY': 0.939243, p-value: 0.156903
	Early neoplasia 'EIF1AY': 0.877818, p-value: 0.006232
	Cancer  'EIF1AY': 0.974394, p-value: 0.792493
	Normal neoplasia 'EIF1AY': 0.978073, p-value: 0.857876
	Early neoplasia 'EIF1AY': 0.903949, p-value: 0.022385
	Cancer  'EIF1AY': 0.972090, p-value: 0.738878
	Normal neoplasia 'EIF1AY': 0.930889, p-value: 0.102052
	Early neoplasia 'EIF1AY': 0.874539, p-value: 0.005342
	Cancer  'EIF1AY': 0.740525, p-value: 0.000049
	Normal neoplasia 'EIF1AY': 0.957510, p-value: 0.390472
	Early neoplasia 'EIF1AY': 0.894320, p-value: 0.013829
	Cancer  'EIF1AY': 0.955676, p-value: 0.381825
	Normal neoplasia 'EIF1AY': 0.900527, p-value: 0.022085
	Early neoplasia 'EIF1AY': 0.955197, p-value: 0.327123
	Cancer  'EIF1AY': 0.965928, p-value: 0.592601
	Normal neoplasia 'EIF1AY': 0.969098, p-value: 0.644761
	Early neoplasia 'EIF1AY': 0.963318, p-value: 0.484422
	Cancer  'EIF1AY': 0.984660, p-value: 0.968339
	Normal neoplasia 'EIF1AY': 0.973659, p-value: 0.757

	Normal neoplasia 'EIF1AY': 0.691832, p-value: 0.000008
	Early neoplasia 'EIF1AY': 0.965739, p-value: 0.540042
	Cancer  'EIF1AY': 0.959664, p-value: 0.456609
	Normal neoplasia 'EIF1AY': 0.952121, p-value: 0.300988
	Early neoplasia 'EIF1AY': 0.960395, p-value: 0.422400
	Cancer  'EIF1AY': 0.944524, p-value: 0.224511
	Normal neoplasia 'EIF1AY': 0.729046, p-value: 0.000025
	Early neoplasia 'EIF1AY': 0.956940, p-value: 0.356909
	Cancer  'EIF1AY': 0.964182, p-value: 0.552751
	Normal neoplasia 'EIF1AY': 0.899388, p-value: 0.020891
	Early neoplasia 'EIF1AY': 0.955464, p-value: 0.331554
	Cancer  'EIF1AY': 0.974594, p-value: 0.797007
	Normal neoplasia 'EIF1AY': 0.919123, p-value: 0.055860
	Early neoplasia 'EIF1AY': 0.954918, p-value: 0.322563
	Cancer  'EIF1AY': 0.970242, p-value: 0.694797
	Normal neoplasia 'EIF1AY': 0.862192, p-value: 0.003713
	Early neoplasia 'EIF1AY': 0.921649, p-value: 0.055869
	Cancer  'EIF1AY': 0.976632, p-value: 0.841591
	Normal neoplasia 'EIF1AY': 0.945863, p-value: 0.220

	Early neoplasia 'EIF1AY': 0.966290, p-value: 0.553184
	Cancer  'EIF1AY': 0.707551, p-value: 0.000018
	Normal neoplasia 'EIF1AY': 0.933728, p-value: 0.118126
	Early neoplasia 'EIF1AY': 0.951182, p-value: 0.266537
	Cancer  'EIF1AY': 0.962662, p-value: 0.519181
	Normal neoplasia 'EIF1AY': 0.881325, p-value: 0.008837
	Early neoplasia 'EIF1AY': 0.858486, p-value: 0.002563
	Cancer  'EIF1AY': 0.614314, p-value: 0.000001
	Normal neoplasia 'EIF1AY': 0.488487, p-value: 0.000000
	Early neoplasia 'EIF1AY': 0.715522, p-value: 0.000012
	Cancer  'EIF1AY': 0.783251, p-value: 0.000209
	Normal neoplasia 'EIF1AY': 0.794122, p-value: 0.000236
	Early neoplasia 'EIF1AY': 0.927837, p-value: 0.077479
	Cancer  'EIF1AY': 0.949466, p-value: 0.285248
	Normal neoplasia 'EIF1AY': 0.936433, p-value: 0.135784
	Early neoplasia 'EIF1AY': 0.930463, p-value: 0.089076
	Cancer  'EIF1AY': 0.977064, p-value: 0.850604
	Normal neoplasia 'EIF1AY': 0.933280, p-value: 0.115428
	Early neoplasia 'EIF1AY': 0.924091, p-value: 0.0635

	Early neoplasia 'EIF1AY': 0.970637, p-value: 0.661258
	Cancer  'EIF1AY': 0.932873, p-value: 0.126171
	Normal neoplasia 'EIF1AY': 0.843745, p-value: 0.001677
	Early neoplasia 'EIF1AY': 0.967148, p-value: 0.573939
	Cancer  'EIF1AY': 0.919904, p-value: 0.066305
	Normal neoplasia 'EIF1AY': 0.954053, p-value: 0.330835
	Early neoplasia 'EIF1AY': 0.963244, p-value: 0.482787
	Cancer  'EIF1AY': 0.967360, p-value: 0.626085
	Normal neoplasia 'EIF1AY': 0.985890, p-value: 0.975557
	Early neoplasia 'EIF1AY': 0.981187, p-value: 0.907581
	Cancer  'EIF1AY': 0.963886, p-value: 0.546129
	Normal neoplasia 'EIF1AY': 0.947410, p-value: 0.237952
	Early neoplasia 'EIF1AY': 0.935143, p-value: 0.114287
	Cancer  'EIF1AY': 0.958934, p-value: 0.442182
	Normal neoplasia 'EIF1AY': 0.976902, p-value: 0.832648
	Early neoplasia 'EIF1AY': 0.946009, p-value: 0.203498
	Cancer  'EIF1AY': 0.952602, p-value: 0.331039
	Normal neoplasia 'EIF1AY': 0.966609, p-value: 0.584446
	Early neoplasia 'EIF1AY': 0.977067, p-value: 0.8215

	Early neoplasia 'EIF1AY': 0.814295, p-value: 0.000398
	Cancer  'EIF1AY': 0.925353, p-value: 0.086818
	Normal neoplasia 'EIF1AY': 0.973547, p-value: 0.754321
	Early neoplasia 'EIF1AY': 0.973399, p-value: 0.731663
	Cancer  'EIF1AY': 0.965013, p-value: 0.571557
	Normal neoplasia 'EIF1AY': 0.970718, p-value: 0.684748
	Early neoplasia 'EIF1AY': 0.944665, p-value: 0.189572
	Cancer  'EIF1AY': 0.984198, p-value: 0.963805
	Normal neoplasia 'EIF1AY': 0.983135, p-value: 0.945856
	Early neoplasia 'EIF1AY': 0.974462, p-value: 0.758430
	Cancer  'EIF1AY': 0.973510, p-value: 0.772186
	Normal neoplasia 'EIF1AY': 0.967530, p-value: 0.606533
	Early neoplasia 'EIF1AY': 0.960903, p-value: 0.432757
	Cancer  'EIF1AY': 0.890038, p-value: 0.015844
	Normal neoplasia 'EIF1AY': 0.949899, p-value: 0.269591
	Early neoplasia 'EIF1AY': 0.976018, p-value: 0.796660
	Cancer  'EIF1AY': 0.947177, p-value: 0.255469
	Normal neoplasia 'EIF1AY': 0.947933, p-value: 0.244296
	Early neoplasia 'EIF1AY': 0.960097, p-value: 0.4164

	Early neoplasia 'EIF1AY': 0.986484, p-value: 0.977354
	Cancer  'EIF1AY': 0.940722, p-value: 0.186237
	Normal neoplasia 'EIF1AY': 0.961885, p-value: 0.477454
	Early neoplasia 'EIF1AY': 0.927595, p-value: 0.076489
	Cancer  'EIF1AY': 0.887475, p-value: 0.014076
	Normal neoplasia 'EIF1AY': 0.864774, p-value: 0.004163
	Early neoplasia 'EIF1AY': 0.898981, p-value: 0.017434
	Cancer  'EIF1AY': 0.955875, p-value: 0.385313
	Normal neoplasia 'EIF1AY': 0.947176, p-value: 0.235149
	Early neoplasia 'EIF1AY': 0.955294, p-value: 0.328733
	Cancer  'EIF1AY': 0.969934, p-value: 0.687428
	Normal neoplasia 'EIF1AY': 0.797105, p-value: 0.000264
	Early neoplasia 'EIF1AY': 0.950617, p-value: 0.258876
	Cancer  'EIF1AY': 0.977200, p-value: 0.853384
	Normal neoplasia 'EIF1AY': 0.913557, p-value: 0.042156
	Early neoplasia 'EIF1AY': 0.981778, p-value: 0.917925
	Cancer  'EIF1AY': 0.906460, p-value: 0.034425
	Normal neoplasia 'EIF1AY': 0.965815, p-value: 0.565656
	Early neoplasia 'EIF1AY': 0.922958, p-value: 0.0598

	Cancer  'EIF1AY': 0.975028, p-value: 0.806766
	Normal neoplasia 'EIF1AY': 0.976079, p-value: 0.814154
	Early neoplasia 'EIF1AY': 0.968242, p-value: 0.600880
	Cancer  'EIF1AY': 0.862544, p-value: 0.004633
	Normal neoplasia 'EIF1AY': 0.647181, p-value: 0.000002
	Early neoplasia 'EIF1AY': 0.780872, p-value: 0.000111
	Cancer  'EIF1AY': 0.706830, p-value: 0.000017
	Normal neoplasia 'EIF1AY': 0.958227, p-value: 0.403843
	Early neoplasia 'EIF1AY': 0.959507, p-value: 0.404746
	Cancer  'EIF1AY': 0.966579, p-value: 0.607743
	Normal neoplasia 'EIF1AY': 0.976419, p-value: 0.821876
	Early neoplasia 'EIF1AY': 0.984298, p-value: 0.955011
	Cancer  'EIF1AY': 0.925198, p-value: 0.086151
	Normal neoplasia 'EIF1AY': 0.945845, p-value: 0.219840
	Early neoplasia 'EIF1AY': 0.935170, p-value: 0.114452
	Cancer  'EIF1AY': 0.835289, p-value: 0.001489
	Normal neoplasia 'EIF1AY': 0.937514, p-value: 0.143557
	Early neoplasia 'EIF1AY': 0.888395, p-value: 0.010344
	Cancer  'EIF1AY': 0.797713, p-value: 0.000352
	Norm

	Normal neoplasia 'EIF1AY': 0.912552, p-value: 0.040079
	Early neoplasia 'EIF1AY': 0.991508, p-value: 0.998421
	Cancer  'EIF1AY': 0.940524, p-value: 0.184431
	Normal neoplasia 'EIF1AY': 0.950901, p-value: 0.283371
	Early neoplasia 'EIF1AY': 0.939723, p-value: 0.145868
	Cancer  'EIF1AY': 0.934767, p-value: 0.138631
	Normal neoplasia 'EIF1AY': 0.953820, p-value: 0.327119
	Early neoplasia 'EIF1AY': 0.962607, p-value: 0.468807
	Cancer  'EIF1AY': 0.903563, p-value: 0.029957
	Normal neoplasia 'EIF1AY': 0.756988, p-value: 0.000063
	Early neoplasia 'EIF1AY': 0.928692, p-value: 0.081072
	Cancer  'EIF1AY': 0.938849, p-value: 0.169767
	Normal neoplasia 'EIF1AY': 0.928352, p-value: 0.089562
	Early neoplasia 'EIF1AY': 0.961061, p-value: 0.436016
	Cancer  'EIF1AY': 0.958374, p-value: 0.431330
	Normal neoplasia 'EIF1AY': 0.952184, p-value: 0.301916
	Early neoplasia 'EIF1AY': 0.974810, p-value: 0.767082
	Cancer  'EIF1AY': 0.965544, p-value: 0.583712
	Normal neoplasia 'EIF1AY': 0.719599, p-value: 0.000

	Early neoplasia 'EIF1AY': 0.957210, p-value: 0.361722
	Cancer  'EIF1AY': 0.768887, p-value: 0.000127
	Normal neoplasia 'EIF1AY': 0.959801, p-value: 0.434448
	Early neoplasia 'EIF1AY': 0.942855, p-value: 0.172264
	Cancer  'EIF1AY': 0.959574, p-value: 0.454810
	Normal neoplasia 'EIF1AY': 0.823569, p-value: 0.000734
	Early neoplasia 'EIF1AY': 0.920547, p-value: 0.052726
	Cancer  'EIF1AY': 0.820893, p-value: 0.000844
	Normal neoplasia 'EIF1AY': 0.953037, p-value: 0.314844
	Early neoplasia 'EIF1AY': 0.959873, p-value: 0.411955
	Cancer  'EIF1AY': 0.915086, p-value: 0.052331
	Normal neoplasia 'EIF1AY': 0.972313, p-value: 0.724152
	Early neoplasia 'EIF1AY': 0.973515, p-value: 0.734602
	Cancer  'EIF1AY': 0.962058, p-value: 0.506160
	Normal neoplasia 'EIF1AY': 0.850035, p-value: 0.002189
	Early neoplasia 'EIF1AY': 0.937494, p-value: 0.129538
	Cancer  'EIF1AY': 0.943596, p-value: 0.214534
	Normal neoplasia 'EIF1AY': 0.931998, p-value: 0.108050
	Early neoplasia 'EIF1AY': 0.943625, p-value: 0.1794

	Early neoplasia 'EIF1AY': 0.954776, p-value: 0.320252
	Cancer  'EIF1AY': 0.972319, p-value: 0.744288
	Normal neoplasia 'EIF1AY': 0.903240, p-value: 0.025226
	Early neoplasia 'EIF1AY': 0.939432, p-value: 0.143626
	Cancer  'EIF1AY': 0.868766, p-value: 0.006072
	Normal neoplasia 'EIF1AY': 0.987495, p-value: 0.986649
	Early neoplasia 'EIF1AY': 0.893240, p-value: 0.013112
	Cancer  'EIF1AY': 0.958655, p-value: 0.436749
	Normal neoplasia 'EIF1AY': 0.946085, p-value: 0.222536
	Early neoplasia 'EIF1AY': 0.983089, p-value: 0.938694
	Cancer  'EIF1AY': 0.966445, p-value: 0.604607
	Normal neoplasia 'EIF1AY': 0.968824, p-value: 0.638032
	Early neoplasia 'EIF1AY': 0.965806, p-value: 0.541622
	Cancer  'EIF1AY': 0.978551, p-value: 0.880056
	Normal neoplasia 'EIF1AY': 0.940637, p-value: 0.168541
	Early neoplasia 'EIF1AY': 0.966084, p-value: 0.548242
	Cancer  'EIF1AY': 0.947312, p-value: 0.257135
	Normal neoplasia 'EIF1AY': 0.979308, p-value: 0.882774
	Early neoplasia 'EIF1AY': 0.937549, p-value: 0.1299

	Early neoplasia 'EIF1AY': 0.892815, p-value: 0.012840
	Cancer  'EIF1AY': 0.935316, p-value: 0.142465
	Normal neoplasia 'EIF1AY': 0.820312, p-value: 0.000645
	Early neoplasia 'EIF1AY': 0.946563, p-value: 0.209511
	Cancer  'EIF1AY': 0.821459, p-value: 0.000863
	Normal neoplasia 'EIF1AY': 0.873432, p-value: 0.006146
	Early neoplasia 'EIF1AY': 0.970044, p-value: 0.646187
	Cancer  'EIF1AY': 0.949304, p-value: 0.283048
	Normal neoplasia 'EIF1AY': 0.778385, p-value: 0.000133
	Early neoplasia 'EIF1AY': 0.956748, p-value: 0.353517
	Cancer  'EIF1AY': 0.966557, p-value: 0.607240
	Normal neoplasia 'EIF1AY': 0.965392, p-value: 0.555778
	Early neoplasia 'EIF1AY': 0.988953, p-value: 0.992118
	Cancer  'EIF1AY': 0.945460, p-value: 0.235017
	Normal neoplasia 'EIF1AY': 0.938790, p-value: 0.153294
	Early neoplasia 'EIF1AY': 0.972451, p-value: 0.707555
	Cancer  'EIF1AY': 0.972038, p-value: 0.737639
	Normal neoplasia 'EIF1AY': 0.985720, p-value: 0.974126
	Early neoplasia 'EIF1AY': 0.969671, p-value: 0.6367

	Cancer  'EIF1AY': 0.969625, p-value: 0.680020
	Normal neoplasia 'EIF1AY': 0.950880, p-value: 0.283071
	Early neoplasia 'EIF1AY': 0.989137, p-value: 0.992832
	Cancer  'EIF1AY': 0.977364, p-value: 0.856743
	Normal neoplasia 'EIF1AY': 0.668605, p-value: 0.000004
	Early neoplasia 'EIF1AY': 0.920927, p-value: 0.053789
	Cancer  'EIF1AY': 0.966506, p-value: 0.606028
	Normal neoplasia 'EIF1AY': 0.943968, p-value: 0.199839
	Early neoplasia 'EIF1AY': 0.938224, p-value: 0.134676
	Cancer  'EIF1AY': 0.966789, p-value: 0.612657
	Normal neoplasia 'EIF1AY': 0.942397, p-value: 0.184443
	Early neoplasia 'EIF1AY': 0.940709, p-value: 0.153714
	Cancer  'EIF1AY': 0.959521, p-value: 0.453770
	Normal neoplasia 'EIF1AY': 0.920155, p-value: 0.058871
	Early neoplasia 'EIF1AY': 0.954446, p-value: 0.314962
	Cancer  'EIF1AY': 0.958172, p-value: 0.427472
	Normal neoplasia 'EIF1AY': 0.970635, p-value: 0.682695
	Early neoplasia 'EIF1AY': 0.972150, p-value: 0.699883
	Cancer  'EIF1AY': 0.972457, p-value: 0.747548
	Norm

	Normal neoplasia 'EIF1AY': 0.971216, p-value: 0.697071
	Early neoplasia 'EIF1AY': 0.955951, p-value: 0.339740
	Cancer  'EIF1AY': 0.968322, p-value: 0.648907
	Normal neoplasia 'EIF1AY': 0.926135, p-value: 0.079924
	Early neoplasia 'EIF1AY': 0.948675, p-value: 0.234015
	Cancer  'EIF1AY': 0.963836, p-value: 0.545008
	Normal neoplasia 'EIF1AY': 0.965270, p-value: 0.552926
	Early neoplasia 'EIF1AY': 0.968120, p-value: 0.597852
	Cancer  'EIF1AY': 0.938431, p-value: 0.166284
	Normal neoplasia 'EIF1AY': 0.981733, p-value: 0.925515
	Early neoplasia 'EIF1AY': 0.967878, p-value: 0.591853
	Cancer  'EIF1AY': 0.971411, p-value: 0.722743
	Normal neoplasia 'EIF1AY': 0.978100, p-value: 0.858429
	Early neoplasia 'EIF1AY': 0.957313, p-value: 0.363556
	Cancer  'EIF1AY': 0.963163, p-value: 0.530123
	Normal neoplasia 'EIF1AY': 0.938875, p-value: 0.153963
	Early neoplasia 'EIF1AY': 0.967018, p-value: 0.570759
	Cancer  'EIF1AY': 0.933037, p-value: 0.127201
	Normal neoplasia 'EIF1AY': 0.945699, p-value: 0.218

	Normal neoplasia 'EIF1AY': 0.947649, p-value: 0.240827
	Early neoplasia 'EIF1AY': 0.957295, p-value: 0.363245
	Cancer  'EIF1AY': 0.950734, p-value: 0.303053
	Normal neoplasia 'EIF1AY': 0.965807, p-value: 0.565474
	Early neoplasia 'EIF1AY': 0.935610, p-value: 0.117165
	Cancer  'EIF1AY': 0.945018, p-value: 0.229994
	Normal neoplasia 'EIF1AY': 0.956841, p-value: 0.378290
	Early neoplasia 'EIF1AY': 0.929145, p-value: 0.083049
	Cancer  'EIF1AY': 0.973746, p-value: 0.777648
	Normal neoplasia 'EIF1AY': 0.899618, p-value: 0.021126
	Early neoplasia 'EIF1AY': 0.649113, p-value: 0.000002
	Cancer  'EIF1AY': 0.756437, p-value: 0.000083
	Normal neoplasia 'EIF1AY': 0.949975, p-value: 0.270604
	Early neoplasia 'EIF1AY': 0.980817, p-value: 0.900812
	Cancer  'EIF1AY': 0.917358, p-value: 0.058498
	Normal neoplasia 'EIF1AY': 0.958900, p-value: 0.416726
	Early neoplasia 'EIF1AY': 0.915338, p-value: 0.040163
	Cancer  'EIF1AY': 0.911509, p-value: 0.043950
	Normal neoplasia 'EIF1AY': 0.952246, p-value: 0.302

	Cancer  'EIF1AY': 0.970669, p-value: 0.705018
	Normal neoplasia 'EIF1AY': 0.934665, p-value: 0.123967
	Early neoplasia 'EIF1AY': 0.946043, p-value: 0.203859
	Cancer  'EIF1AY': 0.964074, p-value: 0.550327
	Normal neoplasia 'EIF1AY': 0.934071, p-value: 0.120230
	Early neoplasia 'EIF1AY': 0.965860, p-value: 0.542925
	Cancer  'EIF1AY': 0.966606, p-value: 0.608378
	Normal neoplasia 'EIF1AY': 0.891137, p-value: 0.014032
	Early neoplasia 'EIF1AY': 0.864737, p-value: 0.003398
	Cancer  'EIF1AY': 0.790032, p-value: 0.000267
	Normal neoplasia 'EIF1AY': 0.971429, p-value: 0.702340
	Early neoplasia 'EIF1AY': 0.980767, p-value: 0.899871
	Cancer  'EIF1AY': 0.966224, p-value: 0.599463
	Normal neoplasia 'EIF1AY': 0.958613, p-value: 0.411194
	Early neoplasia 'EIF1AY': 0.966890, p-value: 0.567641
	Cancer  'EIF1AY': 0.954006, p-value: 0.353500
	Normal neoplasia 'EIF1AY': 0.981934, p-value: 0.928629
	Early neoplasia 'EIF1AY': 0.946273, p-value: 0.206344
	Cancer  'EIF1AY': 0.948345, p-value: 0.270295
	Norm

	Cancer  'EIF1AY': 0.976789, p-value: 0.844898
	Normal neoplasia 'EIF1AY': 0.950571, p-value: 0.278759
	Early neoplasia 'EIF1AY': 0.702397, p-value: 0.000008
	Cancer  'EIF1AY': 0.971844, p-value: 0.733027
	Normal neoplasia 'EIF1AY': 0.955453, p-value: 0.354018
	Early neoplasia 'EIF1AY': 0.963378, p-value: 0.485767
	Cancer  'EIF1AY': 0.954574, p-value: 0.362931
	Normal neoplasia 'EIF1AY': 0.965111, p-value: 0.549262
	Early neoplasia 'EIF1AY': 0.986973, p-value: 0.981115
	Cancer  'EIF1AY': 0.937913, p-value: 0.162070
	Normal neoplasia 'EIF1AY': 0.973109, p-value: 0.743673
	Early neoplasia 'EIF1AY': 0.979067, p-value: 0.866006
	Cancer  'EIF1AY': 0.980863, p-value: 0.920318
	Normal neoplasia 'EIF1AY': 0.940596, p-value: 0.168188
	Early neoplasia 'EIF1AY': 0.916533, p-value: 0.042740
	Cancer  'EIF1AY': 0.898155, p-value: 0.023165
	Normal neoplasia 'EIF1AY': 0.928314, p-value: 0.089390
	Early neoplasia 'EIF1AY': 0.970072, p-value: 0.646895
	Cancer  'EIF1AY': 0.969313, p-value: 0.672558
	Norm

	Cancer  'EIF1AY': 0.984838, p-value: 0.969990
	Normal neoplasia 'EIF1AY': 0.931625, p-value: 0.105992
	Early neoplasia 'EIF1AY': 0.886392, p-value: 0.009387
	Cancer  'EIF1AY': 0.852371, p-value: 0.003004
	Normal neoplasia 'EIF1AY': 0.850552, p-value: 0.002238
	Early neoplasia 'EIF1AY': 0.890989, p-value: 0.011740
	Cancer  'EIF1AY': 0.779271, p-value: 0.000182
	Normal neoplasia 'EIF1AY': 0.913784, p-value: 0.042639
	Early neoplasia 'EIF1AY': 0.944373, p-value: 0.186670
	Cancer  'EIF1AY': 0.653585, p-value: 0.000004
	Normal neoplasia 'EIF1AY': 0.982269, p-value: 0.933685
	Early neoplasia 'EIF1AY': 0.958324, p-value: 0.382098
	Cancer  'EIF1AY': 0.860551, p-value: 0.004252
	Normal neoplasia 'EIF1AY': 0.923819, p-value: 0.070979
	Early neoplasia 'EIF1AY': 0.918514, p-value: 0.047398
	Cancer  'EIF1AY': 0.939955, p-value: 0.179312
	Normal neoplasia 'EIF1AY': 0.946865, p-value: 0.231487
	Early neoplasia 'EIF1AY': 0.975404, p-value: 0.781728
	Cancer  'EIF1AY': 0.966582, p-value: 0.607814
	Norm

	Normal neoplasia 'EIF1AY': 0.832046, p-value: 0.001033
	Early neoplasia 'EIF1AY': 0.929630, p-value: 0.085217
	Cancer  'EIF1AY': 0.904733, p-value: 0.031684
	Normal neoplasia 'EIF1AY': 0.976778, p-value: 0.829902
	Early neoplasia 'EIF1AY': 0.973222, p-value: 0.727189
	Cancer  'EIF1AY': 0.966981, p-value: 0.617175
	Normal neoplasia 'EIF1AY': 0.978689, p-value: 0.870523
	Early neoplasia 'EIF1AY': 0.952182, p-value: 0.280626
	Cancer  'EIF1AY': 0.975019, p-value: 0.806569
	Normal neoplasia 'EIF1AY': 0.954931, p-value: 0.345232
	Early neoplasia 'EIF1AY': 0.974476, p-value: 0.758777
	Cancer  'EIF1AY': 0.917405, p-value: 0.058632
	Normal neoplasia 'EIF1AY': 0.904228, p-value: 0.026485
	Early neoplasia 'EIF1AY': 0.951985, p-value: 0.277802
	Cancer  'EIF1AY': 0.883586, p-value: 0.011779
	Normal neoplasia 'EIF1AY': 0.865334, p-value: 0.004268
	Early neoplasia 'EIF1AY': 0.968203, p-value: 0.599905
	Cancer  'EIF1AY': 0.943562, p-value: 0.214174
	Normal neoplasia 'EIF1AY': 0.922340, p-value: 0.065

	Cancer  'EIF1AY': 0.963633, p-value: 0.540482
	Normal neoplasia 'EIF1AY': 0.967657, p-value: 0.609589
	Early neoplasia 'EIF1AY': 0.886942, p-value: 0.009640
	Cancer  'EIF1AY': 0.971761, p-value: 0.731073
	Normal neoplasia 'EIF1AY': 0.933975, p-value: 0.119635
	Early neoplasia 'EIF1AY': 0.973132, p-value: 0.724903
	Cancer  'EIF1AY': 0.871868, p-value: 0.006961
	Normal neoplasia 'EIF1AY': 0.924423, p-value: 0.073208
	Early neoplasia 'EIF1AY': 0.883478, p-value: 0.008158
	Cancer  'EIF1AY': 0.954799, p-value: 0.366737
	Normal neoplasia 'EIF1AY': 0.951059, p-value: 0.285600
	Early neoplasia 'EIF1AY': 0.911398, p-value: 0.032754
	Cancer  'EIF1AY': 0.965624, p-value: 0.585555
	Normal neoplasia 'EIF1AY': 0.975261, p-value: 0.795243
	Early neoplasia 'EIF1AY': 0.919639, p-value: 0.050273
	Cancer  'EIF1AY': 0.938632, p-value: 0.167953
	Normal neoplasia 'EIF1AY': 0.962708, p-value: 0.495200
	Early neoplasia 'EIF1AY': 0.964726, p-value: 0.516345
	Cancer  'EIF1AY': 0.745239, p-value: 0.000057
	Norm

	Early neoplasia 'EIF1AY': 0.906746, p-value: 0.025804
	Cancer  'EIF1AY': 0.938010, p-value: 0.162851
	Normal neoplasia 'EIF1AY': 0.815954, p-value: 0.000543
	Early neoplasia 'EIF1AY': 0.941906, p-value: 0.163809
	Cancer  'EIF1AY': 0.974094, p-value: 0.785641
	Normal neoplasia 'EIF1AY': 0.941085, p-value: 0.172461
	Early neoplasia 'EIF1AY': 0.940387, p-value: 0.151105
	Cancer  'EIF1AY': 0.765695, p-value: 0.000114
	Normal neoplasia 'EIF1AY': 0.852306, p-value: 0.002413
	Early neoplasia 'EIF1AY': 0.944371, p-value: 0.186658
	Cancer  'EIF1AY': 0.893995, p-value: 0.019050
	Normal neoplasia 'EIF1AY': 0.955613, p-value: 0.356757
	Early neoplasia 'EIF1AY': 0.972218, p-value: 0.701617
	Cancer  'EIF1AY': 0.978523, p-value: 0.879523
	Normal neoplasia 'EIF1AY': 0.937786, p-value: 0.145583
	Early neoplasia 'EIF1AY': 0.956866, p-value: 0.355592
	Cancer  'EIF1AY': 0.978189, p-value: 0.873125
	Normal neoplasia 'EIF1AY': 0.964819, p-value: 0.542519
	Early neoplasia 'EIF1AY': 0.959466, p-value: 0.4039

	Normal neoplasia 'EIF1AY': 0.846410, p-value: 0.001877
	Early neoplasia 'EIF1AY': 0.976593, p-value: 0.810386
	Cancer  'EIF1AY': 0.947698, p-value: 0.261978
	Normal neoplasia 'EIF1AY': 0.973912, p-value: 0.763153
	Early neoplasia 'EIF1AY': 0.954747, p-value: 0.319789
	Cancer  'EIF1AY': 0.909629, p-value: 0.040118
	Normal neoplasia 'EIF1AY': 0.948504, p-value: 0.251412
	Early neoplasia 'EIF1AY': 0.975564, p-value: 0.785638
	Cancer  'EIF1AY': 0.958114, p-value: 0.426370
	Normal neoplasia 'EIF1AY': 0.784406, p-value: 0.000166
	Early neoplasia 'EIF1AY': 0.920105, p-value: 0.051517
	Cancer  'EIF1AY': 0.906266, p-value: 0.034106
	Normal neoplasia 'EIF1AY': 0.913991, p-value: 0.043086
	Early neoplasia 'EIF1AY': 0.709802, p-value: 0.000010
	Cancer  'EIF1AY': 0.910416, p-value: 0.041678
	Normal neoplasia 'EIF1AY': 0.821134, p-value: 0.000666
	Early neoplasia 'EIF1AY': 0.811839, p-value: 0.000361
	Cancer  'EIF1AY': 0.898031, p-value: 0.023030
	Normal neoplasia 'EIF1AY': 0.957444, p-value: 0.389

	Early neoplasia 'EIF1AY': 0.949643, p-value: 0.246122
	Cancer  'EIF1AY': 0.956560, p-value: 0.397546
	Normal neoplasia 'EIF1AY': 0.965895, p-value: 0.567537
	Early neoplasia 'EIF1AY': 0.962098, p-value: 0.457817
	Cancer  'EIF1AY': 0.968699, p-value: 0.657897
	Normal neoplasia 'EIF1AY': 0.982831, p-value: 0.941744
	Early neoplasia 'EIF1AY': 0.949119, p-value: 0.239497
	Cancer  'EIF1AY': 0.927378, p-value: 0.095997
	Normal neoplasia 'EIF1AY': 0.980707, p-value: 0.908528
	Early neoplasia 'EIF1AY': 0.971352, p-value: 0.679498
	Cancer  'EIF1AY': 0.897062, p-value: 0.022000
	Normal neoplasia 'EIF1AY': 0.955257, p-value: 0.350700
	Early neoplasia 'EIF1AY': 0.895543, p-value: 0.014692
	Cancer  'EIF1AY': 0.975828, p-value: 0.824409
	Normal neoplasia 'EIF1AY': 0.956533, p-value: 0.372806
	Early neoplasia 'EIF1AY': 0.964674, p-value: 0.515151
	Cancer  'EIF1AY': 0.932666, p-value: 0.124877
	Normal neoplasia 'EIF1AY': 0.989609, p-value: 0.995248
	Early neoplasia 'EIF1AY': 0.918456, p-value: 0.0472

	Normal neoplasia 'EIF1AY': 0.875566, p-value: 0.006775
	Early neoplasia 'EIF1AY': 0.873878, p-value: 0.005180
	Cancer  'EIF1AY': 0.965159, p-value: 0.574887
	Normal neoplasia 'EIF1AY': 0.988733, p-value: 0.992416
	Early neoplasia 'EIF1AY': 0.983765, p-value: 0.948165
	Cancer  'EIF1AY': 0.972286, p-value: 0.743502
	Normal neoplasia 'EIF1AY': 0.938309, p-value: 0.149548
	Early neoplasia 'EIF1AY': 0.980761, p-value: 0.899770
	Cancer  'EIF1AY': 0.979134, p-value: 0.890896
	Normal neoplasia 'EIF1AY': 0.822420, p-value: 0.000701
	Early neoplasia 'EIF1AY': 0.854190, p-value: 0.002117
	Cancer  'EIF1AY': 0.901038, p-value: 0.026558
	Normal neoplasia 'EIF1AY': 0.967838, p-value: 0.613993
	Early neoplasia 'EIF1AY': 0.969390, p-value: 0.629648
	Cancer  'EIF1AY': 0.950673, p-value: 0.302171
	Normal neoplasia 'EIF1AY': 0.947964, p-value: 0.244682
	Early neoplasia 'EIF1AY': 0.964573, p-value: 0.512807
	Cancer  'EIF1AY': 0.930524, p-value: 0.112252
	Normal neoplasia 'EIF1AY': 0.868139, p-value: 0.004

	Normal neoplasia 'EIF1AY': 0.986652, p-value: 0.981367
	Early neoplasia 'EIF1AY': 0.965380, p-value: 0.531569
	Cancer  'EIF1AY': 0.950695, p-value: 0.302498
	Normal neoplasia 'EIF1AY': 0.928358, p-value: 0.089590
	Early neoplasia 'EIF1AY': 0.986469, p-value: 0.977230
	Cancer  'EIF1AY': 0.928642, p-value: 0.102223
	Normal neoplasia 'EIF1AY': 0.917778, p-value: 0.052175
	Early neoplasia 'EIF1AY': 0.973562, p-value: 0.735797
	Cancer  'EIF1AY': 0.968487, p-value: 0.652836
	Normal neoplasia 'EIF1AY': 0.945351, p-value: 0.214407
	Early neoplasia 'EIF1AY': 0.962184, p-value: 0.459666
	Cancer  'EIF1AY': 0.972402, p-value: 0.746238
	Normal neoplasia 'EIF1AY': 0.960552, p-value: 0.449624
	Early neoplasia 'EIF1AY': 0.956327, p-value: 0.346183
	Cancer  'EIF1AY': 0.899113, p-value: 0.024238
	Normal neoplasia 'EIF1AY': 0.886306, p-value: 0.011158
	Early neoplasia 'EIF1AY': 0.955529, p-value: 0.332628
	Cancer  'EIF1AY': 0.890922, p-value: 0.016508
	Normal neoplasia 'EIF1AY': 0.938843, p-value: 0.153

	Normal neoplasia 'EIF1AY': 0.967727, p-value: 0.611306
	Early neoplasia 'EIF1AY': 0.945208, p-value: 0.195087
	Cancer  'EIF1AY': 0.950392, p-value: 0.298164
	Normal neoplasia 'EIF1AY': 0.978497, p-value: 0.866622
	Early neoplasia 'EIF1AY': 0.962504, p-value: 0.466562
	Cancer  'EIF1AY': 0.972803, p-value: 0.755683
	Normal neoplasia 'EIF1AY': 0.959175, p-value: 0.422066
	Early neoplasia 'EIF1AY': 0.959763, p-value: 0.409769
	Cancer  'EIF1AY': 0.933904, p-value: 0.132810
	Normal neoplasia 'EIF1AY': 0.944954, p-value: 0.210127
	Early neoplasia 'EIF1AY': 0.989437, p-value: 0.993893
	Cancer  'EIF1AY': 0.967405, p-value: 0.627154
	Normal neoplasia 'EIF1AY': 0.922712, p-value: 0.067074
	Early neoplasia 'EIF1AY': 0.911103, p-value: 0.032259
	Cancer  'EIF1AY': 0.959697, p-value: 0.457268
	Normal neoplasia 'EIF1AY': 0.980451, p-value: 0.904033
	Early neoplasia 'EIF1AY': 0.866104, p-value: 0.003617
	Cancer  'EIF1AY': 0.974024, p-value: 0.784033
	Normal neoplasia 'EIF1AY': 0.921359, p-value: 0.062

	Normal neoplasia 'EIF1AY': 0.951281, p-value: 0.288748
	Early neoplasia 'EIF1AY': 0.869542, p-value: 0.004235
	Cancer  'EIF1AY': 0.978128, p-value: 0.871926
	Normal neoplasia 'EIF1AY': 0.960948, p-value: 0.457769
	Early neoplasia 'EIF1AY': 0.957227, p-value: 0.362012
	Cancer  'EIF1AY': 0.920476, p-value: 0.068204
	Normal neoplasia 'EIF1AY': 0.976982, p-value: 0.834428
	Early neoplasia 'EIF1AY': 0.972885, p-value: 0.718612
	Cancer  'EIF1AY': 0.950060, p-value: 0.293467
	Normal neoplasia 'EIF1AY': 0.895874, p-value: 0.017616
	Early neoplasia 'EIF1AY': 0.934685, p-value: 0.111528
	Cancer  'EIF1AY': 0.936553, p-value: 0.151492
	Normal neoplasia 'EIF1AY': 0.965467, p-value: 0.557530
	Early neoplasia 'EIF1AY': 0.975101, p-value: 0.774273
	Cancer  'EIF1AY': 0.986075, p-value: 0.980001
	Normal neoplasia 'EIF1AY': 0.972861, p-value: 0.737606
	Early neoplasia 'EIF1AY': 0.977769, p-value: 0.837628
	Cancer  'EIF1AY': 0.963134, p-value: 0.529471
	Normal neoplasia 'EIF1AY': 0.847079, p-value: 0.001

	Early neoplasia 'EIF1AY': 0.964498, p-value: 0.511102
	Cancer  'EIF1AY': 0.974550, p-value: 0.796015
	Normal neoplasia 'EIF1AY': 0.966884, p-value: 0.591002
	Early neoplasia 'EIF1AY': 0.956460, p-value: 0.348496
	Cancer  'EIF1AY': 0.933966, p-value: 0.133219
	Normal neoplasia 'EIF1AY': 0.954349, p-value: 0.335628
	Early neoplasia 'EIF1AY': 0.942560, p-value: 0.169595
	Cancer  'EIF1AY': 0.941633, p-value: 0.194801
	Normal neoplasia 'EIF1AY': 0.971403, p-value: 0.701690
	Early neoplasia 'EIF1AY': 0.815277, p-value: 0.000413
	Cancer  'EIF1AY': 0.974366, p-value: 0.791837
	Normal neoplasia 'EIF1AY': 0.799251, p-value: 0.000286
	Early neoplasia 'EIF1AY': 0.983883, p-value: 0.949717
	Cancer  'EIF1AY': 0.944554, p-value: 0.224838
	Normal neoplasia 'EIF1AY': 0.934811, p-value: 0.124902
	Early neoplasia 'EIF1AY': 0.930002, p-value: 0.086918
	Cancer  'EIF1AY': 0.963613, p-value: 0.540041
	Normal neoplasia 'EIF1AY': 0.907664, p-value: 0.031395
	Early neoplasia 'EIF1AY': 0.951367, p-value: 0.2691

	Normal neoplasia 'EIF1AY': 0.929819, p-value: 0.096582
	Early neoplasia 'EIF1AY': 0.962754, p-value: 0.472015
	Cancer  'EIF1AY': 0.954477, p-value: 0.361316
	Normal neoplasia 'EIF1AY': 0.923399, p-value: 0.069472
	Early neoplasia 'EIF1AY': 0.954375, p-value: 0.313835
	Cancer  'EIF1AY': 0.917883, p-value: 0.060026
	Normal neoplasia 'EIF1AY': 0.898278, p-value: 0.019792
	Early neoplasia 'EIF1AY': 0.968893, p-value: 0.617135
	Cancer  'EIF1AY': 0.960516, p-value: 0.473872
	Normal neoplasia 'EIF1AY': 0.837905, p-value: 0.001314
	Early neoplasia 'EIF1AY': 0.942106, p-value: 0.165554
	Cancer  'EIF1AY': 0.929303, p-value: 0.105637
	Normal neoplasia 'EIF1AY': 0.953479, p-value: 0.321722
	Early neoplasia 'EIF1AY': 0.974540, p-value: 0.760373
	Cancer  'EIF1AY': 0.956450, p-value: 0.395562
	Normal neoplasia 'EIF1AY': 0.955513, p-value: 0.355044
	Early neoplasia 'EIF1AY': 0.972109, p-value: 0.698825
	Cancer  'EIF1AY': 0.952797, p-value: 0.334081
	Normal neoplasia 'EIF1AY': 0.974254, p-value: 0.771

	Early neoplasia 'EIF1AY': 0.973672, p-value: 0.738580
	Cancer  'EIF1AY': 0.965730, p-value: 0.588019
	Normal neoplasia 'EIF1AY': 0.833022, p-value: 0.001075
	Early neoplasia 'EIF1AY': 0.669047, p-value: 0.000003
	Cancer  'EIF1AY': 0.910938, p-value: 0.042748
	Normal neoplasia 'EIF1AY': 0.972256, p-value: 0.722739
	Early neoplasia 'EIF1AY': 0.942552, p-value: 0.169523
	Cancer  'EIF1AY': 0.910536, p-value: 0.041922
	Normal neoplasia 'EIF1AY': 0.900014, p-value: 0.021539
	Early neoplasia 'EIF1AY': 0.942658, p-value: 0.170475
	Cancer  'EIF1AY': 0.985327, p-value: 0.974252
	Normal neoplasia 'EIF1AY': 0.978078, p-value: 0.857973
	Early neoplasia 'EIF1AY': 0.975363, p-value: 0.780719
	Cancer  'EIF1AY': 0.975491, p-value: 0.817038
	Normal neoplasia 'EIF1AY': 0.979406, p-value: 0.884675
	Early neoplasia 'EIF1AY': 0.949563, p-value: 0.245096
	Cancer  'EIF1AY': 0.963953, p-value: 0.547612
	Normal neoplasia 'EIF1AY': 0.960295, p-value: 0.444384
	Early neoplasia 'EIF1AY': 0.955934, p-value: 0.3394

	Cancer  'EIF1AY': 0.961070, p-value: 0.485308
	Normal neoplasia 'EIF1AY': 0.950440, p-value: 0.276945
	Early neoplasia 'EIF1AY': 0.974162, p-value: 0.750921
	Cancer  'EIF1AY': 0.946727, p-value: 0.249946
	Normal neoplasia 'EIF1AY': 0.936642, p-value: 0.137258
	Early neoplasia 'EIF1AY': 0.916423, p-value: 0.042496
	Cancer  'EIF1AY': 0.961391, p-value: 0.492029
	Normal neoplasia 'EIF1AY': 0.944640, p-value: 0.206797
	Early neoplasia 'EIF1AY': 0.899623, p-value: 0.018003
	Cancer  'EIF1AY': 0.944974, p-value: 0.229503
	Normal neoplasia 'EIF1AY': 0.970179, p-value: 0.671413
	Early neoplasia 'EIF1AY': 0.970696, p-value: 0.662772
	Cancer  'EIF1AY': 0.982753, p-value: 0.947243
	Normal neoplasia 'EIF1AY': 0.918126, p-value: 0.053104
	Early neoplasia 'EIF1AY': 0.783287, p-value: 0.000121
	Cancer  'EIF1AY': 0.959820, p-value: 0.459747
	Normal neoplasia 'EIF1AY': 0.938174, p-value: 0.148516
	Early neoplasia 'EIF1AY': 0.959576, p-value: 0.406094
	Cancer  'EIF1AY': 0.950811, p-value: 0.304167
	Norm

	Cancer  'EIF1AY': 0.930205, p-value: 0.110486
	Normal neoplasia 'EIF1AY': 0.947473, p-value: 0.238707
	Early neoplasia 'EIF1AY': 0.962909, p-value: 0.475403
	Cancer  'EIF1AY': 0.869839, p-value: 0.006365
	Normal neoplasia 'EIF1AY': 0.876598, p-value: 0.007103
	Early neoplasia 'EIF1AY': 0.948747, p-value: 0.234904
	Cancer  'EIF1AY': 0.886586, p-value: 0.013513
	Normal neoplasia 'EIF1AY': 0.962464, p-value: 0.489900
	Early neoplasia 'EIF1AY': 0.950096, p-value: 0.251976
	Cancer  'EIF1AY': 0.969755, p-value: 0.683136
	Normal neoplasia 'EIF1AY': 0.954517, p-value: 0.338375
	Early neoplasia 'EIF1AY': 0.844870, p-value: 0.001410
	Cancer  'EIF1AY': 0.930895, p-value: 0.114341
	Normal neoplasia 'EIF1AY': 0.668804, p-value: 0.000004
	Early neoplasia 'EIF1AY': 0.959144, p-value: 0.397694
	Cancer  'EIF1AY': 0.790162, p-value: 0.000268
	Normal neoplasia 'EIF1AY': 0.822270, p-value: 0.000697
	Early neoplasia 'EIF1AY': 0.628117, p-value: 0.000001
	Cancer  'EIF1AY': 0.813619, p-value: 0.000638
	Norm

	Cancer  'EIF1AY': 0.844815, p-value: 0.002194
	Normal neoplasia 'EIF1AY': 0.950611, p-value: 0.279311
	Early neoplasia 'EIF1AY': 0.922207, p-value: 0.057534
	Cancer  'EIF1AY': 0.964201, p-value: 0.553171
	Normal neoplasia 'EIF1AY': 0.955053, p-value: 0.347256
	Early neoplasia 'EIF1AY': 0.961445, p-value: 0.444003
	Cancer  'EIF1AY': 0.989396, p-value: 0.995444
	Normal neoplasia 'EIF1AY': 0.955150, p-value: 0.348892
	Early neoplasia 'EIF1AY': 0.917904, p-value: 0.045910
	Cancer  'EIF1AY': 0.948023, p-value: 0.266124
	Normal neoplasia 'EIF1AY': 0.971003, p-value: 0.691791
	Early neoplasia 'EIF1AY': 0.928038, p-value: 0.078308
	Cancer  'EIF1AY': 0.971000, p-value: 0.712921
	Normal neoplasia 'EIF1AY': 0.897302, p-value: 0.018876
	Early neoplasia 'EIF1AY': 0.974174, p-value: 0.751218
	Cancer  'EIF1AY': 0.977216, p-value: 0.853726
	Normal neoplasia 'EIF1AY': 0.965778, p-value: 0.564785
	Early neoplasia 'EIF1AY': 0.989947, p-value: 0.995435
	Cancer  'EIF1AY': 0.966443, p-value: 0.604572
	Norm

	Early neoplasia 'EIF1AY': 0.912467, p-value: 0.034610
	Cancer  'EIF1AY': 0.971740, p-value: 0.730566
	Normal neoplasia 'EIF1AY': 0.958531, p-value: 0.409626
	Early neoplasia 'EIF1AY': 0.924096, p-value: 0.063558
	Cancer  'EIF1AY': 0.952240, p-value: 0.325445
	Normal neoplasia 'EIF1AY': 0.920272, p-value: 0.059224
	Early neoplasia 'EIF1AY': 0.907494, p-value: 0.026808
	Cancer  'EIF1AY': 0.946868, p-value: 0.251668
	Normal neoplasia 'EIF1AY': 0.955252, p-value: 0.350613
	Early neoplasia 'EIF1AY': 0.966150, p-value: 0.549829
	Cancer  'EIF1AY': 0.973262, p-value: 0.766425
	Normal neoplasia 'EIF1AY': 0.981802, p-value: 0.926584
	Early neoplasia 'EIF1AY': 0.953409, p-value: 0.298796
	Cancer  'EIF1AY': 0.951606, p-value: 0.315857
	Normal neoplasia 'EIF1AY': 0.967156, p-value: 0.597512
	Early neoplasia 'EIF1AY': 0.960913, p-value: 0.432955
	Cancer  'EIF1AY': 0.865240, p-value: 0.005206
	Normal neoplasia 'EIF1AY': 0.922937, p-value: 0.067850
	Early neoplasia 'EIF1AY': 0.965191, p-value: 0.5271

	Normal neoplasia 'EIF1AY': 0.825575, p-value: 0.000795
	Early neoplasia 'EIF1AY': 0.786479, p-value: 0.000136
	Cancer  'EIF1AY': 0.923191, p-value: 0.077998
	Normal neoplasia 'EIF1AY': 0.973129, p-value: 0.744157
	Early neoplasia 'EIF1AY': 0.963657, p-value: 0.492010
	Cancer  'EIF1AY': 0.936175, p-value: 0.148679
	Normal neoplasia 'EIF1AY': 0.979940, p-value: 0.894748
	Early neoplasia 'EIF1AY': 0.953004, p-value: 0.292685
	Cancer  'EIF1AY': 0.967914, p-value: 0.639206
	Normal neoplasia 'EIF1AY': 0.942820, p-value: 0.188474
	Early neoplasia 'EIF1AY': 0.833791, p-value: 0.000881
	Cancer  'EIF1AY': 0.948362, p-value: 0.270512
	Normal neoplasia 'EIF1AY': 0.990747, p-value: 0.997673
	Early neoplasia 'EIF1AY': 0.943008, p-value: 0.173671
	Cancer  'EIF1AY': 0.968070, p-value: 0.642910
	Normal neoplasia 'EIF1AY': 0.981722, p-value: 0.925329
	Early neoplasia 'EIF1AY': 0.938677, p-value: 0.137966
	Cancer  'EIF1AY': 0.941594, p-value: 0.194428
	Normal neoplasia 'EIF1AY': 0.973487, p-value: 0.752

	Early neoplasia 'EIF1AY': 0.750594, p-value: 0.000038
	Cancer  'EIF1AY': 0.943710, p-value: 0.215741
	Normal neoplasia 'EIF1AY': 0.953341, p-value: 0.319554
	Early neoplasia 'EIF1AY': 0.929186, p-value: 0.083231
	Cancer  'EIF1AY': 0.977436, p-value: 0.858186
	Normal neoplasia 'EIF1AY': 0.978903, p-value: 0.874807
	Early neoplasia 'EIF1AY': 0.954513, p-value: 0.316024
	Cancer  'EIF1AY': 0.948626, p-value: 0.273979
	Normal neoplasia 'EIF1AY': 0.961948, p-value: 0.478806
	Early neoplasia 'EIF1AY': 0.955699, p-value: 0.335476
	Cancer  'EIF1AY': 0.980499, p-value: 0.914486
	Normal neoplasia 'EIF1AY': 0.986883, p-value: 0.982930
	Early neoplasia 'EIF1AY': 0.948426, p-value: 0.230999
	Cancer  'EIF1AY': 0.968808, p-value: 0.660484
	Normal neoplasia 'EIF1AY': 0.960732, p-value: 0.453307
	Early neoplasia 'EIF1AY': 0.976373, p-value: 0.805178
	Cancer  'EIF1AY': 0.965223, p-value: 0.576351
	Normal neoplasia 'EIF1AY': 0.864101, p-value: 0.004040
	Early neoplasia 'EIF1AY': 0.957670, p-value: 0.3700

	Normal neoplasia 'EIF1AY': 0.955745, p-value: 0.359024
	Early neoplasia 'EIF1AY': 0.934920, p-value: 0.112938
	Cancer  'EIF1AY': 0.964084, p-value: 0.550543
	Normal neoplasia 'EIF1AY': 0.883407, p-value: 0.009738
	Early neoplasia 'EIF1AY': 0.964391, p-value: 0.508648
	Cancer  'EIF1AY': 0.965688, p-value: 0.587036
	Normal neoplasia 'EIF1AY': 0.664483, p-value: 0.000004
	Early neoplasia 'EIF1AY': 0.751793, p-value: 0.000039
	Cancer  'EIF1AY': 0.819627, p-value: 0.000803
	Normal neoplasia 'EIF1AY': 0.958174, p-value: 0.402846
	Early neoplasia 'EIF1AY': 0.978375, p-value: 0.851119
	Cancer  'EIF1AY': 0.941621, p-value: 0.194681
	Normal neoplasia 'EIF1AY': 0.918301, p-value: 0.053578
	Early neoplasia 'EIF1AY': 0.903153, p-value: 0.021502
	Cancer  'EIF1AY': 0.903642, p-value: 0.030071
	Normal neoplasia 'EIF1AY': 0.884210, p-value: 0.010112
	Early neoplasia 'EIF1AY': 0.967052, p-value: 0.571604
	Cancer  'EIF1AY': 0.968981, p-value: 0.664621
	Normal neoplasia 'EIF1AY': 0.838722, p-value: 0.001

Проверим гипотезу о равенстве средних в двух независимых выборках. Уровень значимости alpha = 0.05
* H0: средние значения РНК в генах равны
* H1: средние не равны

In [42]:
def genes_diffs(right, left, columns):
    return [scipy.stats.ttest_ind(right[i], left[i], equal_var = False).pvalue for i in columns]

In [60]:
normalEarly_diff_genes = genes_diffs(normal, early_neoplasia, data_columns)
normalEarly_diff_genes_count = 0
for p in range(len(normalEarly_diff_genes)):
    if normalEarly_diff_genes[p] < 0.05:
        normalEarly_diff_genes_count += 1
print(normalEarly_diff_genes_count)
write_answer('normal_early.txt', normalEarly_diff_genes_count)

1575


In [59]:
earlyCancer_diff_genes = genes_diffs(early_neoplasia, cancer, data_columns)
earlyCancer_diff_genes_count = 0
for p in range(len(earlyCancer_diff_genes)):
    if earlyCancer_diff_genes[p] < 0.05:
        earlyCancer_diff_genes_count += 1
print(earlyCancer_diff_genes_count)
write_answer('early_cancer.txt', earlyCancer_diff_genes_count)

3490


Видим что для первых групп насчитывается 1575 значимых отличий. Во втором случае - 3490 генов, у которых p-value теста оказался меньше, чем уровень значимости.  

### Часть 2: поправка методом Холма

Теперь применим поправку Холма для получившихся двух наборов достигаемых уровней значимости из первой части. Чтобы избавиться от проблемы, связанной с множественной проверкой, применим поправку Бонферрони, то есть возьмем уровень значимости alpha = 0.05/2.

В качестве ответа к этой части определим количество значимых отличий в каждой группе после того, как произведем коррекцию Холма-Бонферрони.

In [162]:
def column_names(array_of_indexes):
    names = []
    for i in array_of_indexes:
        names.append(list(data)[2:][i])
    return names
    
def fold_change(sample1, sample2):
    mean1 = statistics.mean(sample1)
    mean2 = statistics.mean(sample2)
    if mean1 < mean2:
        return mean2/mean1
    else:
        return mean1/mean2

In [166]:
reject_n_e, pvalues_normalEarly_holm, alphac_n_e_sidak, alphac_n_e_bonf = smm.multipletests(
    normalEarly_diff_genes, method = 'holm')
column_indexes = np.where(pvalues_normalEarly_holm < 0.025)

In [167]:
counter = 0
for i in column_names(column_indexes[0]):
    if fold_change(data[data['Diagnosis'] == 'normal'][i], data[data['Diagnosis'] == 'early neoplasia'][i]) > 1.5:
        counter += 1
print(counter)
write_answer('fold1.txt', counter)

2


In [169]:
reject_n_e, pvalues_normalEarly_holm, alphac_n_e_sidak, alphac_n_e_bonf = smm.multipletests(
    earlyCancer_diff_genes, method = 'holm')
column_indexes1 = np.where(pvalues_normalEarly_holm < 0.025)

In [171]:
counter1 = 0
for i in column_names(column_indexes1[0]):
    if fold_change(data[data['Diagnosis'] == 'cancer'][i], data[data['Diagnosis'] == 'early neoplasia'][i]) > 1.5:
        counter1 += 1
print(counter1)
write_answer('fold2.txt', counter1)

77


Видим, что, применив поправку Холма, количество значимых отличий снизилось: в первой группе стало равно 2, во второй - 77. Причем это числа уже с учетом практической значимости - для каждого значимого изменения посчитали переменную fold change и выписали те изменения, значения которых больше, чем 1.5.

### Часть 3: поправка методом Бенджамини-Хохберга

В этом случае мы повторим действия второй части, но будем использовать уже метод Бенджамини-Хохберга.

В итоге получим количество значимых отличий в каждой группе после того, как произведена коррекция Бенджамини-Хохберга с учетом fold change. 

In [173]:
reject_n_e, pvalues_normalEarly_holm, alphac_n_e_sidak, alphac_n_e_bonf = smm.multipletests(
    normalEarly_diff_genes, method = 'fdr_bh')
column_indexes = np.where(pvalues_normalEarly_holm < 0.025)

counter2 = 0
for i in column_names(column_indexes[0]):
    if fold_change(data[data['Diagnosis'] == 'normal'][i], data[data['Diagnosis'] == 'early neoplasia'][i]) > 1.5:
        counter2 += 1
print(counter)
write_answer('fold3.txt', counter2)

4


In [178]:
reject_n_e, pvalues_normalEarly_holm, alphac_n_e_sidak, alphac_n_e_bonf = smm.multipletests(
    earlyCancer_diff_genes, method = 'fdr_bh')
column_indexes1 = np.where(pvalues_normalEarly_holm < 0.025)

counter3 = 0
for i in column_names(column_indexes1[0]):
    if fold_change(data[data['Diagnosis'] == 'cancer'][i], data[data['Diagnosis'] == 'early neoplasia'][i]) > 1.5:
        counter3 += 1
print(counter3)
write_answer('fold4.txt', counter3)

524


Как видим, в первой группе результат - 4, во второй результат равен 524. Стоит помнить, что данный метод коррекции допускает больше ошибок первого рода и имеет большую мощность, относитьльно метода Холма.
Резюмируя, в ходе проверке гипотез мы определили те гены, активность которых у людей в разных стадиях заболевания отличается статистически значимо.